<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-07-24" />

<title>Merge samples total cells</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TCRM_Christina</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Merge samples total cells</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">24 July 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>TCRM_Christina/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200518code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200518)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200518code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200518)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeTCRMChristinatree5474212f9c9e5e0f43b7547bb44bc7a7c465a5c1targetblank5474212a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/TCRM_Christina/tree/5474212f9c9e5e0f43b7547bb44bc7a7c465a5c1" target="_blank">5474212</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeTCRMChristinatree5474212f9c9e5e0f43b7547bb44bc7a7c465a5c1targetblank5474212a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/TCRM_Christina/tree/5474212f9c9e5e0f43b7547bb44bc7a7c465a5c1" target="_blank">5474212</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/cellphonedb/
    Ignored:    data/data_old/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/GSEAendothelial.Rmd
    Untracked:  analysis/analyseCPDBout.Rmd
    Untracked:  analysis/createCBDBinput.Rmd
    Untracked:  analysis/mergeTotalCells_MuHeart_plusTreat.Rmd

Unstaged changes:
    Modified:   analysis/GSEAfibroblasts.Rmd
    Modified:   analysis/GSEAtotalCells.Rmd
    Modified:   analysis/characterizeTotalCellsHumanHeart.Rmd
    Modified:   metadata_totalCell2.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/characterizeTotalCellsHumanHeart.Rmd</code>) and HTML (<code>docs/characterizeTotalCellsHumanHeart.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5474212f9c9e5e0f43b7547bb44bc7a7c465a5c1/analysis/characterizeTotalCellsHumanHeart.Rmd" target="_blank">5474212</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-18
</td>
<td>
add GSEA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/5474212f9c9e5e0f43b7547bb44bc7a7c465a5c1/docs/characterizeTotalCellsHumanHeart.html" target="_blank">5474212</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-18
</td>
<td>
add GSEA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/analysis/characterizeTotalCellsHumanHeart.Rmd" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
<td>
remove cont cluster total cells
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/characterizeTotalCellsHumanHeart.html" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
<td>
remove cont cluster total cells
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/analysis/characterizeTotalCellsHumanHeart.Rmd" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
<td>
characterize total cells with additional samples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/characterizeTotalCellsHumanHeart.html" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
<td>
characterize total cells with additional samples
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(pheatmap)
  library(ggpubr)
})</code></pre>
</div>
<div id="set-dir" class="section level2">
<h2>set dir</h2>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(file = paste0(basedir,
                              &quot;/data/MuHeart_totalCells_plusTreated_integrated&quot;,
                              &quot;_seurat.rds&quot;))
Idents(seurat) &lt;- seurat$integrated_snn_res.0.4


## remove two contaminating clusters and rerun dim reductions
## cluster 1 (dying cells, a lot of ribosomal genes)
## cluster 17 (erythrocytes (Fech, Bpgm, Snca))
seurat &lt;- subset(seurat, idents = c(&quot;1&quot;, &quot;17&quot;), invert=T)

DefaultAssay(object = seurat) &lt;- &quot;integrated&quot;
seurat &lt;- RunPCA(object = seurat, npcs = 20, verbose = FALSE)
seurat &lt;- RunTSNE(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)
seurat &lt;- RunUMAP(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)
DefaultAssay(object = seurat) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- pal_igv()(length(levels(seurat)))
colGrp &lt;- pal_jama()(length(unique(seurat$grp)))
colAge &lt;- pal_uchicago()(length(unique(seurat$age)))
colStype &lt;- pal_aaas()(length(unique(seurat$seqType)))
colDat &lt;- c(pal_npg()(10),
            pal_futurama()(10))[1:(length(unique(seurat$dataset)))]
colTreat &lt;- pal_jama()(length(unique(seurat$treatment)))

  
names(colPal) &lt;- levels(seurat)
names(colGrp) &lt;- unique(seurat$grp)
names(colAge) &lt;- unique(seurat$age)
names(colStype) &lt;- unique(seurat$seqType)
names(colDat) &lt;- unique(seurat$dataset)
names(colTreat) &lt;- unique(seurat$treatment)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="clusters" class="section level3">
<h3>clusters</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-1">
Past versions of “vis cluster-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis cluster-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis cluster-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="group" class="section level3">
<h3>group</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;grp&quot;, cols=colGrp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20grp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-grp-1">
Past versions of “vis grp-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-grp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis grp-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis grp-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;dataset&quot;, cols=colDat)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20sample-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-sample-1">
Past versions of “vis sample-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis sample-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis sample-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="age" class="section level3">
<h3>Age</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;age&quot;, cols=colAge)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20age-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-age-1">
Past versions of “vis age-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-age-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis age-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis age-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="seqtype" class="section level3">
<h3>SeqType</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;seqType&quot;, cols=colStype)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20seqType-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-seqType-1">
Past versions of “vis seqType-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-seqType-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis seqType-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis seqType-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment" class="section level3">
<h3>Treatment</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;treatment&quot;, cols=colTreat)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="assign-labels" class="section level2 tabset">
<h2 class="tabset">assign labels</h2>
<pre class="r"><code>## -------------------------------------------------------------------------- ##
## cluster 9 re-embedding (Tcells + NKcells mixed)
seuratSub &lt;- subset(seurat, integrated_snn_res.0.4 == &quot;9&quot;)
DefaultAssay(object = seuratSub) &lt;- &quot;integrated&quot;
seuratSub &lt;- ScaleData(object = seuratSub, verbose = FALSE,
                        features = rownames(seuratSub))
seuratSub &lt;- RunPCA(object = seuratSub, npcs = 20, verbose = FALSE)
seuratSub &lt;- RunTSNE(object = seuratSub, reduction = &quot;pca&quot;, dims = 1:20)
seuratSub &lt;- RunUMAP(object = seuratSub, reduction = &quot;pca&quot;, dims = 1:20)

seuratSub &lt;- FindNeighbors(object = seuratSub, reduction = &quot;pca&quot;, dims = 1:20)
res &lt;- c(0.1,0.25,0.4)
for(i in 1:length(res)){
  seuratSub &lt;- FindClusters(object = seuratSub, resolution = res[i],
                             seuratSub = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1593
Number of edges: 50746

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9629
Number of communities: 5
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1593
Number of edges: 50746

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9272
Number of communities: 7
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1593
Number of edges: 50746

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9048
Number of communities: 9
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>Tcells &lt;- colnames(seuratSub)[seuratSub$integrated_snn_res.0.1 %in% c(&quot;2&quot;, &quot;4&quot;)]
NKcells &lt;- colnames(seuratSub)[seuratSub$integrated_snn_res.0.1 %in% c(&quot;0&quot;, &quot;1&quot;, &quot;3&quot;)]


## -------------------------------------------------------------------------- ##


seurat$label &lt;- &quot;other&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;4&quot;))] &lt;- &quot;Macrophages&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;12&quot;))] &lt;- &quot;Bcells&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;6&quot;))] &lt;- &quot;Tcells&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;14&quot;))] &lt;- &quot;Neutrophils&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;11&quot;))] &lt;- &quot;pDCsDCs&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;15&quot;))] &lt;- &quot;proliferating&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;0&quot;, &quot;13&quot;))] &lt;-&quot;Fibroblasts&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;2&quot;, &quot;19&quot;))] &lt;- &quot;Cardiomyocytes&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;3&quot;, &quot;5&quot;))] &lt;- &quot;Endothelial&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;7&quot;, &quot;16&quot;, &quot;18&quot;))] &lt;- &quot;Perivascular&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;10&quot;))] &lt;- &quot;endocardialEndothelial&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;20&quot;))] &lt;- &quot;capEndothelial&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;8&quot;))] &lt;- &quot;InfMonos&quot;
seurat$label[which(colnames(seurat) %in% NKcells)] &lt;- &quot;NKcells&quot;
seurat$label[which(colnames(seurat) %in% Tcells)] &lt;- &quot;Tcells&quot;

colLab &lt;- pal_igv()(length(unique(seurat$label)))
names(colLab) &lt;- unique(seurat$label)</code></pre>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-1">
Past versions of “vis label-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-label-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis label-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis label-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label-split-by-seqtype" class="section level3">
<h3>label split by seqType</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;seqType&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20label%20split%20seq-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-split-seq-1">
Past versions of “vis label split seq-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-label-split-seq-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis label split seq-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis label split seq-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label-split-by-grp" class="section level3">
<h3>label split by grp</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;grp&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20label%20split%20grp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-split-grp-1">
Past versions of “vis label split grp-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-label-split-grp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/6b0c6db4ced5b91fd1abb26f2bd985608d1c9537/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis label split grp-1.png" target="_blank">6b0c6db</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/f1373873f92e392ac69a01aa399a68a438c09ec1/docs/figure/characterizeTotalCellsHumanHeart.Rmd/vis label split grp-1.png" target="_blank">f137387</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label-split-by-treat" class="section level3">
<h3>label split by treat</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;treatment&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20label%20split%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cnt-tab" class="section level3">
<h3>cnt Tab</h3>
<pre class="r"><code>table(seurat$label, seurat$seqType)</code></pre>
<pre><code>                        
                            sc    sn
  Bcells                   577    81
  capEndothelial           112     6
  Cardiomyocytes            65  7227
  endocardialEndothelial    61  1090
  Endothelial             6788  2954
  Fibroblasts            12573    80
  InfMonos                1913   100
  Macrophages             2951  2241
  Neutrophils              546     0
  NKcells                  593   520
  pDCsDCs                  705    98
  Perivascular             911  2111
  proliferating            449    85
  Tcells                  2745    48</code></pre>
<pre class="r"><code>table(seurat$label, seurat$grp)</code></pre>
<pre><code>                        
                         control TCRM
  Bcells                      66  592
  capEndothelial              56   62
  Cardiomyocytes             450 6842
  endocardialEndothelial     209  942
  Endothelial               3069 6673
  Fibroblasts               2874 9779
  InfMonos                   159 1854
  Macrophages                498 4694
  Neutrophils                 21  525
  NKcells                     48 1065
  pDCsDCs                     18  785
  Perivascular               755 2267
  proliferating               33  501
  Tcells                     185 2608</code></pre>
<pre class="r"><code>table(seurat$label, seurat$treatment)</code></pre>
<pre><code>                        
                         14D102 isotype   no
  Bcells                     42     294  322
  capEndothelial             30      17   71
  Cardiomyocytes           2732    2405 2155
  endocardialEndothelial    198     346  607
  Endothelial              1940    1843 5959
  Fibroblasts              3797    2473 6383
  InfMonos                  100     283 1630
  Macrophages              1204    1406 2582
  Neutrophils                62      69  415
  NKcells                   102     231  780
  pDCsDCs                   153     279  371
  Perivascular              736     617 1669
  proliferating              46     145  343
  Tcells                    319     877 1597</code></pre>
</div>
</div>
<div id="vis-bmps-all-incl-treated" class="section level2 tabset">
<h2 class="tabset">vis BMPs all incl treated</h2>
<div id="bmp2" class="section level3">
<h3>Bmp2</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000027358.Bmp2&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmp2%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bmp4" class="section level3">
<h3>Bmp4</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000021835.Bmp4&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmp4%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bmpr1a" class="section level3">
<h3>Bmpr1a</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000021796.Bmpr1a&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmpr1a%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bmpr2" class="section level3">
<h3>Bmpr2</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000067336.Bmpr2&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmpr2%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="save-integrated-objects-all" class="section level2">
<h2>save integrated objects all</h2>
<pre class="r"><code>saveRDS(seurat, file = paste0(basedir,
                      &quot;/data/MuHeart_totalCellsPlusTreated_integratedLabeled&quot;, 
                      &quot;_seurat.rds&quot;))</code></pre>
</div>
<div id="only-untreated" class="section level2">
<h2>only untreated</h2>
<pre class="r"><code>seurat &lt;- subset(seurat, treatment == &quot;no&quot;)</code></pre>
</div>
<div id="vis-data-1" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="label-1" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20label%20wo%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="label-split-by-seqtype-1" class="section level3">
<h3>label split by seqType</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;seqType&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20label%20split%20seq%20wo%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="label-split-by-grp-1" class="section level3">
<h3>label split by grp</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;grp&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20label%20split%20grp%20wo%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cw-de-genes" class="section level2 tabset">
<h2 class="tabset">cw DE genes</h2>
<div id="weeks" class="section level3">
<h3>4 weeks</h3>
<pre class="r"><code>seuratSub &lt;- subset(seurat, age == &quot;4wk&quot;)
Idents(seuratSub) &lt;- seuratSub$grp
grpVec &lt;- unique(seuratSub$label)

clustDE &lt;- lapply(grpVec, function(grp){
  grpSub &lt;- unique(seuratSub$label)[which(
    unique(seuratSub$label)==grp)]
  seuratSub2 &lt;- subset(seuratSub, label == grpSub)
  DEgenes &lt;-FindAllMarkers(seuratSub2, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.1)
  if(nrow(DEgenes)&gt;1){
    DEgenes &lt;- DEgenes %&gt;% filter(p_val_adj&lt;0.01) %&gt;%
      mutate(group=paste0(grp, &quot;_&quot;, cluster))
    }
})

names(clustDE) &lt;- grpVec

clustDE_Dat4 &lt;- data.frame(do.call(&quot;rbind&quot;, clustDE))</code></pre>
</div>
<div id="weeks-1" class="section level3">
<h3>8 weeks</h3>
<pre class="r"><code>seuratSub &lt;- subset(seurat, age == &quot;8wk&quot;)
Idents(seuratSub) &lt;- seuratSub$grp
grpVec &lt;- unique(seuratSub$label)

clustDE &lt;- lapply(grpVec, function(grp){
  grpSub &lt;- unique(seuratSub$label)[which(
    unique(seuratSub$label)==grp)]
  seuratSub2 &lt;- subset(seuratSub, label == grpSub)
  DEgenes &lt;-FindAllMarkers(seuratSub2, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.1)
  if(nrow(DEgenes)&gt;1){
    DEgenes &lt;- DEgenes %&gt;% filter(p_val_adj&lt;0.01) %&gt;%
      mutate(group=paste0(grp, &quot;_&quot;, cluster))
    }
})

names(clustDE) &lt;- grpVec


clustDE_Dat8 &lt;- data.frame(do.call(&quot;rbind&quot;, clustDE))</code></pre>
</div>
</div>
<div id="vis-bmps" class="section level2 tabset">
<h2 class="tabset">vis BMPs</h2>
<div id="bmp2-1" class="section level3">
<h3>Bmp2</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000027358.Bmp2&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmp2%20wo%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bmp4-1" class="section level3">
<h3>Bmp4</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000021835.Bmp4&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmp4%20wo%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bmpr1a-1" class="section level3">
<h3>Bmpr1a</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000021796.Bmpr1a&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmpr1a%20wo%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bmpr2-1" class="section level3">
<h3>Bmpr2</h3>
<pre class="r"><code>p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = &quot;ENSMUSG00000067336.Bmpr2&quot;,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
p</code></pre>
<p><img src="figure/characterizeTotalCellsHumanHeart.Rmd/vis%20Bmpr2%20wo%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="save-integrated-objects" class="section level2">
<h2>save integrated objects</h2>
<pre class="r"><code>saveRDS(seurat, file = paste0(basedir,
                      &quot;/data/MuHeart_totalCells_integratedLabeled_seurat.rds&quot;))

write.table(clustDE_Dat4,
            file=paste0(basedir,
              &quot;/data/MuHeart_totalCells_integratedLabeled_4wks_cwDEGenes.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)

write.table(clustDE_Dat8,
            file=paste0(basedir,
              &quot;/data/MuHeart_totalCells_integratedLabeled_8wks_cwDEGenes.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggpubr_0.4.0                pheatmap_1.0.12            
 [3] ggsci_2.9                   runSeurat3_0.1.0           
 [5] here_1.0.1                  magrittr_2.0.1             
 [7] SeuratObject_4.0.2          Seurat_4.0.5               
 [9] forcats_0.5.1               stringr_1.4.0              
[11] dplyr_1.0.7                 purrr_0.3.4                
[13] readr_2.0.2                 tidyr_1.1.4                
[15] tibble_3.1.6                ggplot2_3.3.5              
[17] tidyverse_1.3.1             SingleCellExperiment_1.14.1
[19] SummarizedExperiment_1.22.0 Biobase_2.52.0             
[21] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[23] IRanges_2.26.0              S4Vectors_0.30.2           
[25] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[27] matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.3.0        workflowr_1.6.2       
  [4] plyr_1.8.6             igraph_1.2.8           lazyeval_0.2.2        
  [7] splines_4.1.0          listenv_0.8.0          scattermore_0.7       
 [10] digest_0.6.28          htmltools_0.5.2        fansi_0.5.0           
 [13] tensor_1.5             cluster_2.1.2          ROCR_1.0-11           
 [16] limma_3.48.3           tzdb_0.2.0             globals_0.14.0        
 [19] modelr_0.1.8           spatstat.sparse_2.0-0  colorspace_2.0-2      
 [22] rvest_1.0.2            ggrepel_0.9.1          haven_2.4.3           
 [25] xfun_0.28              crayon_1.4.2           RCurl_1.98-1.5        
 [28] jsonlite_1.7.2         spatstat.data_2.1-0    survival_3.2-13       
 [31] zoo_1.8-9              glue_1.5.0             polyclip_1.10-0       
 [34] gtable_0.3.0           zlibbioc_1.38.0        XVector_0.32.0        
 [37] leiden_0.3.9           DelayedArray_0.18.0    car_3.0-12            
 [40] future.apply_1.8.1     abind_1.4-5            scales_1.1.1          
 [43] DBI_1.1.1              rstatix_0.7.0          miniUI_0.1.1.1        
 [46] Rcpp_1.0.7             viridisLite_0.4.0      xtable_1.8-4          
 [49] reticulate_1.22        spatstat.core_2.3-1    htmlwidgets_1.5.4     
 [52] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.2        
 [55] ica_1.0-2              farver_2.1.0           pkgconfig_2.0.3       
 [58] uwot_0.1.10            deldir_1.0-6           sass_0.4.0            
 [61] dbplyr_2.1.1           utf8_1.2.2             labeling_0.4.2        
 [64] tidyselect_1.1.1       rlang_0.4.12           reshape2_1.4.4        
 [67] later_1.3.0            munsell_0.5.0          cellranger_1.1.0      
 [70] tools_4.1.0            cli_3.1.0              generics_0.1.1        
 [73] broom_0.7.10           ggridges_0.5.3         evaluate_0.14         
 [76] fastmap_1.1.0          goftest_1.2-3          yaml_2.2.1            
 [79] knitr_1.36             fs_1.5.0               fitdistrplus_1.1-6    
 [82] RANN_2.6.1             nlme_3.1-153           pbapply_1.5-0         
 [85] future_1.23.0          whisker_0.4            mime_0.12             
 [88] xml2_1.3.2             compiler_4.1.0         rstudioapi_0.13       
 [91] plotly_4.10.0          png_0.1-7              ggsignif_0.6.3        
 [94] spatstat.utils_2.2-0   reprex_2.0.1           bslib_0.3.1           
 [97] stringi_1.7.5          highr_0.9              RSpectra_0.16-0       
[100] lattice_0.20-45        Matrix_1.3-4           vctrs_0.3.8           
[103] pillar_1.6.4           lifecycle_1.0.1        spatstat.geom_2.3-0   
[106] lmtest_0.9-39          jquerylib_0.1.4        RcppAnnoy_0.0.19      
[109] data.table_1.14.2      cowplot_1.1.1          bitops_1.0-7          
[112] irlba_2.3.3            httpuv_1.6.3           patchwork_1.1.1       
[115] R6_2.5.1               promises_1.2.0.1       KernSmooth_2.23-20    
[118] gridExtra_2.3          parallelly_1.28.1      codetools_0.2-18      
[121] MASS_7.3-54            assertthat_0.2.1       rprojroot_2.0.2       
[124] withr_2.4.2            sctransform_0.3.2      GenomeInfoDbData_1.2.6
[127] mgcv_1.8-38            hms_1.1.1              rpart_4.1-15          
[130] grid_4.1.0             rmarkdown_2.11         carData_3.0-4         
[133] Rtsne_0.15             git2r_0.28.0           shiny_1.7.1           
[136] lubridate_1.8.0       </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Fri Dec 17 16:27:57 2021&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggpubr_0.4.0                pheatmap_1.0.12            
 [3] ggsci_2.9                   runSeurat3_0.1.0           
 [5] here_1.0.1                  magrittr_2.0.1             
 [7] SeuratObject_4.0.2          Seurat_4.0.5               
 [9] forcats_0.5.1               stringr_1.4.0              
[11] dplyr_1.0.7                 purrr_0.3.4                
[13] readr_2.0.2                 tidyr_1.1.4                
[15] tibble_3.1.6                ggplot2_3.3.5              
[17] tidyverse_1.3.1             SingleCellExperiment_1.14.1
[19] SummarizedExperiment_1.22.0 Biobase_2.52.0             
[21] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[23] IRanges_2.26.0              S4Vectors_0.30.2           
[25] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[27] matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.3.0        workflowr_1.6.2       
  [4] plyr_1.8.6             igraph_1.2.8           lazyeval_0.2.2        
  [7] splines_4.1.0          listenv_0.8.0          scattermore_0.7       
 [10] digest_0.6.28          htmltools_0.5.2        fansi_0.5.0           
 [13] tensor_1.5             cluster_2.1.2          ROCR_1.0-11           
 [16] limma_3.48.3           tzdb_0.2.0             globals_0.14.0        
 [19] modelr_0.1.8           spatstat.sparse_2.0-0  colorspace_2.0-2      
 [22] rvest_1.0.2            ggrepel_0.9.1          haven_2.4.3           
 [25] xfun_0.28              crayon_1.4.2           RCurl_1.98-1.5        
 [28] jsonlite_1.7.2         spatstat.data_2.1-0    survival_3.2-13       
 [31] zoo_1.8-9              glue_1.5.0             polyclip_1.10-0       
 [34] gtable_0.3.0           zlibbioc_1.38.0        XVector_0.32.0        
 [37] leiden_0.3.9           DelayedArray_0.18.0    car_3.0-12            
 [40] future.apply_1.8.1     abind_1.4-5            scales_1.1.1          
 [43] DBI_1.1.1              rstatix_0.7.0          miniUI_0.1.1.1        
 [46] Rcpp_1.0.7             viridisLite_0.4.0      xtable_1.8-4          
 [49] reticulate_1.22        spatstat.core_2.3-1    htmlwidgets_1.5.4     
 [52] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.2        
 [55] ica_1.0-2              farver_2.1.0           pkgconfig_2.0.3       
 [58] uwot_0.1.10            deldir_1.0-6           sass_0.4.0            
 [61] dbplyr_2.1.1           utf8_1.2.2             labeling_0.4.2        
 [64] tidyselect_1.1.1       rlang_0.4.12           reshape2_1.4.4        
 [67] later_1.3.0            munsell_0.5.0          cellranger_1.1.0      
 [70] tools_4.1.0            cli_3.1.0              generics_0.1.1        
 [73] broom_0.7.10           ggridges_0.5.3         evaluate_0.14         
 [76] fastmap_1.1.0          goftest_1.2-3          yaml_2.2.1            
 [79] knitr_1.36             fs_1.5.0               fitdistrplus_1.1-6    
 [82] RANN_2.6.1             nlme_3.1-153           pbapply_1.5-0         
 [85] future_1.23.0          whisker_0.4            mime_0.12             
 [88] xml2_1.3.2             compiler_4.1.0         rstudioapi_0.13       
 [91] plotly_4.10.0          png_0.1-7              ggsignif_0.6.3        
 [94] spatstat.utils_2.2-0   reprex_2.0.1           bslib_0.3.1           
 [97] stringi_1.7.5          highr_0.9              RSpectra_0.16-0       
[100] lattice_0.20-45        Matrix_1.3-4           vctrs_0.3.8           
[103] pillar_1.6.4           lifecycle_1.0.1        spatstat.geom_2.3-0   
[106] lmtest_0.9-39          jquerylib_0.1.4        RcppAnnoy_0.0.19      
[109] data.table_1.14.2      cowplot_1.1.1          bitops_1.0-7          
[112] irlba_2.3.3            httpuv_1.6.3           patchwork_1.1.1       
[115] R6_2.5.1               promises_1.2.0.1       KernSmooth_2.23-20    
[118] gridExtra_2.3          parallelly_1.28.1      codetools_0.2-18      
[121] MASS_7.3-54            assertthat_0.2.1       rprojroot_2.0.2       
[124] withr_2.4.2            sctransform_0.3.2      GenomeInfoDbData_1.2.6
[127] mgcv_1.8-38            hms_1.1.1              rpart_4.1-15          
[130] grid_4.1.0             rmarkdown_2.11         carData_3.0-4         
[133] Rtsne_0.15             git2r_0.28.0           shiny_1.7.1           
[136] lubridate_1.8.0       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
