<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-07-24" />

<title>DE genes treated versus untreated total cells</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TCRM_Christina</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DE genes treated versus untreated total
cells</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">24 July 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>TCRM_Christina/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200518code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200518)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200518code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200518)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeTCRMChristinatree08365111919cd89ffca2e57676c5daa1900bdffctargetblank0836511a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mluetge/TCRM_Christina/tree/08365111919cd89ffca2e57676c5daa1900bdffc" target="_blank">0836511</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeTCRMChristinatree08365111919cd89ffca2e57676c5daa1900bdffctargetblank0836511a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mluetge/TCRM_Christina/tree/08365111919cd89ffca2e57676c5daa1900bdffc" target="_blank">0836511</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/cellphonedb/
    Ignored:    data/data_old/
    Ignored:    figure/.DS_Store
    Ignored:    output/.DS_Store

Unstaged changes:
    Modified:   analysis/DEgenesTreatment.Rmd
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE Bcells-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE Cardiomyocytes-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE DCs-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE Endothelial-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE Fibroblasts-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE InfMonos-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE Macrophages-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE NKcells-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE Perivascular-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE Tcells-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE endocardialEndothelial-1.pdf
    Modified:   figure/DEgenesTreatment.Rmd/avgHeat cwDE proliferating-1.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DEgenesTreatment.Rmd</code>) and
HTML (<code>docs/DEgenesTreatment.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/analysis/DEgenesTreatment.Rmd" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
<td>
update marker etc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/5c506594387375bcff645a7583c36704005b0a31/docs/DEgenesTreatment.html" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
<td>
update marker etc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/analysis/DEgenesTreatment.Rmd" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
<td>
update analysis total cells treated plus untreated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/DEgenesTreatment.html" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
<td>
update analysis total cells treated plus untreated
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(pheatmap)
  library(ggpubr)
  library(clusterProfiler)
  library(org.Mm.eg.db)
  library(DOSE)
  library(enrichplot)
  library(viridis)
})</code></pre>
</div>
<div id="heatmap-function" class="section level2">
<h2>heatmap function</h2>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;.*_&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
</div>
<div id="sign-plot-funct" class="section level2">
<h2>sign plot funct</h2>
<pre class="r"><code>## adapted from CellMixS
visGroup_adapt &lt;- function (sce,group,dim_red = &quot;TSNE&quot;,col_group=pal_nejm()(8)) 
{
    if (!is(sce, &quot;SingleCellExperiment&quot;)) {
        stop(&quot;Error:&#39;sce&#39; must be a &#39;SingleCellExperiment&#39; object.&quot;)
    }
    if (!group %in% names(colData(sce))) {
        stop(&quot;Error: &#39;group&#39; variable must be in &#39;colData(sce)&#39;&quot;)
    }
    cell_names &lt;- colnames(sce)
    if (!dim_red %in% &quot;TSNE&quot;) {
        if (!dim_red %in% reducedDimNames(sce)) {
            stop(&quot;Please provide a dim_red method listed in reducedDims of sce&quot;)
        }
        red_dim &lt;- as.data.frame(reducedDim(sce, dim_red))
    }
    else {
        if (!&quot;TSNE&quot; %in% reducedDimNames(sce)) {
            if (&quot;logcounts&quot; %in% names(assays(sce))) {
                sce &lt;- runTSNE(sce)
            }
            else {
                sce &lt;- runTSNE(sce, exprs_values = &quot;counts&quot;)
            }
        }
        red_dim &lt;- as.data.frame(reducedDim(sce, &quot;TSNE&quot;))
    }
    colnames(red_dim) &lt;- c(&quot;red_dim1&quot;, &quot;red_dim2&quot;)
    df &lt;- data.frame(sample_id = cell_names, group_var = colData(sce)[, 
        group], red_Dim1 = red_dim$red_dim1, red_Dim2 = red_dim$red_dim2)
    t &lt;- ggplot(df, aes_string(x = &quot;red_Dim1&quot;, y = &quot;red_Dim2&quot;)) + 
        xlab(paste0(dim_red, &quot;_1&quot;)) + ylab(paste0(dim_red, &quot;_2&quot;)) + 
        theme_void() + theme(aspect.ratio = 1,
                             panel.grid.minor = element_blank(), 
        panel.grid.major = element_line(color = &quot;grey&quot;, size = 0.3))
    t_group &lt;- t + geom_point(size = 1.5, alpha = 0.8,
                              aes_string(color = &quot;group_var&quot;)) + 
        guides(color = guide_legend(override.aes = list(size = 1), 
            title = group)) + ggtitle(group)
    if (is.numeric(df$group_var)) {
        t_group &lt;- t_group + scale_color_viridis(option = &quot;D&quot;)
    }
    else {
        t_group &lt;- t_group + scale_color_manual(values = col_group)
    }
    t_group
}</code></pre>
</div>
<div id="set-dir" class="section level2">
<h2>set dir</h2>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(file = paste0(basedir,
                      &quot;/data/MuHeart_totalCellsPlusTreated_integratedLabeled&quot;, 
                      &quot;_seurat.rds&quot;))
Idents(seurat) &lt;- seurat$label
seurat$age[which(seurat$treatment != &quot;no&quot;)] &lt;- &quot;12wk&quot;</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- pal_igv()(length(levels(seurat)))
colGrp &lt;- pal_jama()(length(unique(seurat$grp)))
colAge &lt;- pal_uchicago()(length(unique(seurat$age)))
colStype &lt;- pal_aaas()(length(unique(seurat$seqType)))
colDat &lt;- c(pal_npg()(10),
            pal_futurama()(10))[1:(length(unique(seurat$dataset)))]
colTreat &lt;- pal_jama()(length(unique(seurat$treatment)))

names(colPal) &lt;- levels(seurat)
names(colGrp) &lt;- unique(seurat$grp)
names(colAge) &lt;- unique(seurat$age)
names(colStype) &lt;- unique(seurat$seqType)
names(colDat) &lt;- unique(seurat$dataset)
names(colTreat) &lt;- unique(seurat$treatment)

colLab &lt;- c(&quot;#5050FF&quot;, &quot;#CE3D32&quot;, &quot;#749B58&quot;, &quot;#F0E685&quot;,
            &quot;#466983&quot;, &quot;#BA6388&quot;, &quot;#5DB1DD&quot;, &quot;#802268&quot;,
            &quot;#6BD76B&quot;, &quot;#D595A7&quot;, &quot;#924822&quot;, &quot;#837B8D&quot;,
            &quot;#C75127&quot;, &quot;#D58F5C&quot;, &quot;#7A65A5&quot;, &quot;#E4AF69&quot;)

names(colLab) &lt;- c(&quot;InfMonos&quot;, &quot;Tcells&quot;, &quot;Endothelial&quot;, &quot;Macrophages&quot;,
                   &quot;Fibroblasts&quot;, &quot;DCs&quot;, &quot;NKcells&quot;, &quot;Perivascular&quot;,
                   &quot;Neutrophils&quot;, &quot;Bcells&quot;, &quot;Cardiomyocytes&quot;, &quot;proliferating&quot;,
                   &quot;pDCs&quot;, &quot;endocardialEndothelial&quot;, &quot;Eosinophils&quot;,
                   &quot;GlialCells&quot;)</code></pre>
</div>
<div id="subset-on-12wks-data" class="section level2">
<h2>subset on 12wks data</h2>
<pre class="r"><code>seurat &lt;- subset(seurat, treatment != &quot;no&quot;)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="group" class="section level3">
<h3>group</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;grp&quot;, cols=colGrp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20grp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-grp-1">
Past versions of "vis grp-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-grp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis grp-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis grp-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;dataset&quot;, cols=colDat)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20sample-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-sample-1">
Past versions of "vis sample-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis sample-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis sample-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="age" class="section level3">
<h3>Age</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;age&quot;, cols=colAge)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20age-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-age-1">
Past versions of "vis age-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-age-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis age-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis age-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="seqtype" class="section level3">
<h3>SeqType</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;seqType&quot;, cols=colStype)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20seqType-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-seqType-1">
Past versions of "vis seqType-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-seqType-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis seqType-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis seqType-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;seqType&quot;, cols=colStype)+
  theme_void() </code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20seqType-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-seqType-2">
Past versions of "vis seqType-2.png"
</button>
</p>
<div id="fig-no-spaces-vis-seqType-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis seqType-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis seqType-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="treatment" class="section level3">
<h3>Treatment</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;treatment&quot;, cols=colTreat)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-treat-1">
Past versions of "vis treat-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-treat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis treat-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis treat-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;treatment&quot;, cols=colTreat)+
  theme_void() </code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20treat-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-treat-2">
Past versions of "vis treat-2.png"
</button>
</p>
<div id="fig-no-spaces-vis-treat-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis treat-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis treat-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-1">
Past versions of "vis label-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-label-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis label-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis label-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab)+
  theme_void() </code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20label-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-2">
Past versions of "vis label-2.png"
</button>
</p>
<div id="fig-no-spaces-vis-label-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis label-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis label-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label-split-by-seqtype" class="section level3">
<h3>label split by seqType</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;seqType&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20label%20split%20seq-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-split-seq-1">
Past versions of "vis label split seq-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-label-split-seq-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis label split seq-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis label split seq-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label-split-by-treat" class="section level3">
<h3>label split by treat</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;treatment&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20label%20split%20treat-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-split-treat-1">
Past versions of "vis label split treat-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-label-split-treat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis label split treat-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis label split treat-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;treatment&quot;)+
  theme_void() </code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20label%20split%20treat-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-split-treat-2">
Past versions of "vis label split treat-2.png"
</button>
</p>
<div id="fig-no-spaces-vis-label-split-treat-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis label split treat-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis label split treat-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cnt-tab" class="section level3">
<h3>cnt Tab</h3>
<pre class="r"><code>table(seurat$label, seurat$seqType)</code></pre>
<pre><code>                        
                           sc   sn
  Bcells                  300   40
  Cardiomyocytes            1 5150
  DCs                     377   32
  endocardialEndothelial   12  486
  Endothelial            2196 1555
  Eosinophils              32    0
  Fibroblasts            6271 4564
  GlialCells               48   53
  InfMonos                245   24
  Macrophages            2657  812
  Neutrophils             105    0
  NKcells                 134   42
  pDCs                     58    0
  Perivascular            201 1007
  proliferating           153   46
  Tcells                  489   70</code></pre>
<pre class="r"><code>table(seurat$label, seurat$grp)</code></pre>
<pre><code>                        
                          TCRM
  Bcells                   340
  Cardiomyocytes          5151
  DCs                      409
  endocardialEndothelial   498
  Endothelial             3751
  Eosinophils               32
  Fibroblasts            10835
  GlialCells               101
  InfMonos                 269
  Macrophages             3469
  Neutrophils              105
  NKcells                  176
  pDCs                      58
  Perivascular            1208
  proliferating            199
  Tcells                   559</code></pre>
<pre class="r"><code>table(seurat$label, seurat$treatment)</code></pre>
<pre><code>                        
                         14D102 isotype
  Bcells                     43     297
  Cardiomyocytes           2739    2412
  DCs                       146     263
  endocardialEndothelial    177     321
  Endothelial              1922    1829
  Eosinophils                18      14
  Fibroblasts              6082    4753
  GlialCells                 62      39
  InfMonos                   66     203
  Macrophages              1440    2029
  Neutrophils                44      61
  NKcells                    69     107
  pDCs                       13      45
  Perivascular              655     553
  proliferating              49     150
  Tcells                    137     422</code></pre>
</div>
</div>
<div id="overall-de-genes" class="section level2 tabset">
<h2 class="tabset">overall DE genes</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$treatment
seurat$label_plus_treat &lt;- paste0(seurat$label, &quot;_&quot;, seurat$treatment)
seurat$label &lt;- factor(seurat$label)

overallDE &lt;- FindAllMarkers(object = seurat, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.25,
                                     logfc.threshold = 0.25,
                                     test.use = &quot;wilcox&quot;)


### plot DE genes top 15 avg logFC
DEGenesAll &lt;- overallDE %&gt;% group_by(cluster) %&gt;% 
  mutate(geneID = gene) %&gt;% top_n(15, avg_log2FC) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% 
  filter(nchar(gene)&gt;1)

Idents(seurat) &lt;- seurat$label_plus_treat
ordDat &lt;- expand.grid(levels(seurat$label), unique(seurat$treatment)) %&gt;% 
  mutate(ord=paste0(Var1, &quot;_&quot;, Var2)) %&gt;% arrange(., Var1)
gapC &lt;- seq(from=2, to=nrow(ordDat), by=2)
gapR &lt;- c(15)

pOut &lt;- avgHeatmap(seurat = seurat, selGenes = DEGenesAll,
                  colVecIdent = colLab, 
                  ordVec=ordDat$ord,
                  gapVecR=gapR, gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colTreat)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/overall%20DE-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-overall-DE-1">
Past versions of "overall DE-1.png"
</button>
</p>
<div id="fig-no-spaces-overall-DE-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/overall DE-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/overall DE-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sc-heatmap" class="section level3">
<h3>sc heatmap</h3>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;integrated&quot;
seurat &lt;- ScaleData(seurat, features = rownames(seurat), split.by = &quot;seqType&quot;)

### plot DE genes top 15 avg logFC
DEGenesAll &lt;- overallDE %&gt;% group_by(cluster) %&gt;% 
  top_n(50, avg_log2FC) 

selFeatures &lt;- DEGenesAll %&gt;% mutate(label=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))
DoHeatmap(seurat, features = selFeatures$gene, group.by = &quot;treatment&quot;, 
          group.colors = colTreat, slot = &#39;scale.data&#39;, label = F,
          disp.min = -0.5, disp.max = 1.5) +
  scale_fill_continuous(type = &quot;viridis&quot;) +
  scale_y_discrete(breaks=selFeatures$gene, labels=selFeatures$label)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/sc%20heatmap%20overall%20%20DE-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sc-heatmap-overall--DE-1">
Past versions of "sc heatmap overall DE-1.png"
</button>
</p>
<div id="fig-no-spaces-sc-heatmap-overall--DE-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/sc heatmap overall  DE-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/sc heatmap overall  DE-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DoHeatmap(seurat, features = selFeatures$gene, group.by = &quot;treatment&quot;, 
          group.colors = colTreat, slot = &#39;scale.data&#39;, label = F,
          disp.min = -1, disp.max = 1.5, draw.lines = F) +
  scale_fill_gradientn(colors=colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(30)) +
  scale_y_discrete(breaks=selFeatures$gene, labels=selFeatures$label)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/sc%20heatmap%20overall%20%20DE-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sc-heatmap-overall--DE-2">
Past versions of "sc heatmap overall DE-2.png"
</button>
</p>
<div id="fig-no-spaces-sc-heatmap-overall--DE-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/sc heatmap overall  DE-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/sc heatmap overall  DE-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="clusterprofiler" class="section level3">
<h3>Clusterprofiler</h3>
<pre class="r"><code>grpVec &lt;- unique(seurat$treatment)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Mm.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec</code></pre>
<pre class="r"><code>## table to select pathways
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir, &quot;/data/GSEA/MuHeart_treatedTCRM_OverallDE&quot;,
            &quot;_GO.txt&quot;))
    

selGO &lt;- read_tsv(paste0(basedir,&quot;/data/GSEA/selGO_treatedTCRM_overall.txt&quot;))
GOconsDat &lt;- GOconsDat %&gt;% filter(ID %in% selGO$GOterm) %&gt;% 
  mutate(Grp=ifelse(subset==&quot;control&quot;, &quot;Ctrl&quot;, &quot;TCRM&quot;))

grpVec &lt;- unique(selGO$Grp)
lapply(grpVec, function(grp){
  selGODat &lt;- GOconsDat %&gt;% filter(subset == grp)
  selGODat &lt;- selGODat %&gt;% mutate(qscore=-log(p.adjust, base=10)) 
  p &lt;- ggbarplot(selGODat, x = &quot;Description&quot;, y = &quot;qscore&quot;,
          fill = &quot;subset&quot;,               
          color = &quot;subset&quot;,            
          palette = colTreat,            
          sort.val = &quot;asc&quot;,           
          sort.by.groups = TRUE      
          #x.text.angle = 90           
          ) + 
  rotate()
p
})</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20summary-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-summary-1">
Past versions of "vis summary-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-summary-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis summary-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis summary-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/vis%20summary-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-summary-2">
Past versions of "vis summary-2.png"
</button>
</p>
<div id="fig-no-spaces-vis-summary-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/vis summary-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/vis summary-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cw-de-genes" class="section level2">
<h2>cw DE genes</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$treatment
grpVec &lt;- unique(seurat$label)

clustDE &lt;- lapply(grpVec, function(grp){
  grpSub &lt;- unique(seurat$label)[which(
    unique(seurat$label)==grp)]
  seuratSub &lt;- subset(seurat, label == grpSub)
  DEgenes &lt;-FindAllMarkers(seuratSub, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.1)
  if(nrow(DEgenes)&gt;1){
    DEgenes &lt;- DEgenes %&gt;% filter(p_val_adj&lt;0.01) %&gt;%
      mutate(group=paste0(grp, &quot;_&quot;, cluster)) %&gt;% 
      mutate(geneID=gene) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% 
      filter(nchar(gene)&gt;1)
    }
})

names(clustDE) &lt;- grpVec

clustDE_Dat &lt;- data.frame(do.call(&quot;rbind&quot;, clustDE))</code></pre>
<div id="avg-heatmap-top-cwde-genes" class="section level3 tabset">
<h3 class="tabset">avg Heatmap top cwDE genes</h3>
<div id="fibroblasts" class="section level4">
<h4>Fibroblasts</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE Fibroblasts-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Fibroblasts-1">
Past versions of "avgHeat cwDE Fibroblasts-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Fibroblasts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE Fibroblasts-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE Fibroblasts-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="macrophages" class="section level4">
<h4>Macrophages</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE Macrophages-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Macrophages-1">
Past versions of "avgHeat cwDE Macrophages-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Macrophages-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE Macrophages-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE Macrophages-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="endothelial" class="section level4">
<h4>Endothelial</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE Endothelial-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Endothelial-1">
Past versions of "avgHeat cwDE Endothelial-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Endothelial-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE Endothelial-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE Endothelial-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dcs" class="section level4">
<h4>DCs</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE DCs-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-DCs-1">
Past versions of "avgHeat cwDE DCs-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-DCs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE DCs-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nkcells" class="section level4">
<h4>NKcells</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE NKcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-NKcells-1">
Past versions of "avgHeat cwDE NKcells-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-NKcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE NKcells-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE NKcells-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="perivascular" class="section level4">
<h4>Perivascular</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE Perivascular-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Perivascular-1">
Past versions of "avgHeat cwDE Perivascular-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Perivascular-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE Perivascular-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE Perivascular-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tcells" class="section level4">
<h4>Tcells</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE Tcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Tcells-1">
Past versions of "avgHeat cwDE Tcells-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Tcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE Tcells-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE Tcells-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="bcells" class="section level4">
<h4>Bcells</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE Bcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Bcells-1">
Past versions of "avgHeat cwDE Bcells-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Bcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE Bcells-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proliferating" class="section level4">
<h4>proliferating</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE proliferating-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-proliferating-1">
Past versions of "avgHeat cwDE proliferating-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-proliferating-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE proliferating-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE proliferating-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="infmonos" class="section level4">
<h4>InfMonos</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE InfMonos-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-InfMonos-1">
Past versions of "avgHeat cwDE InfMonos-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-InfMonos-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE InfMonos-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE InfMonos-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="endocardialendothelial" class="section level4">
<h4>endocardialEndothelial</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE endocardialEndothelial-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-endocardialEndothelial-1">
Past versions of "avgHeat cwDE endocardialEndothelial-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-endocardialEndothelial-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE endocardialEndothelial-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE endocardialEndothelial-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cardiomyocytes" class="section level4">
<h4>Cardiomyocytes</h4>
<p><img src="figure/DEgenesTreatment.Rmd/avgHeat cwDE Cardiomyocytes-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Cardiomyocytes-1">
Past versions of "avgHeat cwDE Cardiomyocytes-1.png"
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Cardiomyocytes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/figure/DEgenesTreatment.Rmd/avgHeat cwDE Cardiomyocytes-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/figure/DEgenesTreatment.Rmd/avgHeat cwDE Cardiomyocytes-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="dotplot-with-selgenes" class="section level2">
<h2>Dotplot with selGenes</h2>
<pre class="r"><code>selGenesDat &lt;- data.frame(gene=c(&quot;Bmp2&quot;, &quot;Bmp4&quot;, &quot;Bmpr1a&quot;, &quot;Bmpr1b&quot;, &quot;Bmpr2&quot;))

genes &lt;- data.frame(ensID=rownames(seurat)) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, ensID))
selGenes &lt;- selGenesDat %&gt;% 
  left_join(., genes, by=&quot;gene&quot;)

seurat$label_plus_treat &lt;- paste0(seurat$label, &quot;_&quot;, seurat$treatment)
Idents(seurat) &lt;- seurat$label_plus_treat
clusterDat &lt;- data.frame(ident=levels(seurat)) %&gt;%
  mutate(label=gsub(&quot;_.*$&quot;, &quot;&quot;, ident)) %&gt;% 
  mutate(grp= gsub(&quot;^.*_&quot;, &quot;&quot;, ident))

Idents(seurat) &lt;- seurat$treatment

DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenes$ensID, scale =T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenes$ensID, labels=selGenes$gene) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/dotplot-1.png" width="364.8" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dotplot-1">
Past versions of dotplot-1.png
</button>
</p>
<div id="fig-dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/dotplot-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/dotplot-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenes$ensID, scale =T, dot.scale =10,
        scale.max = 50) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenes$ensID, labels=selGenes$gene) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/dotplot-2.png" width="364.8" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dotplot-2">
Past versions of dotplot-2.png
</button>
</p>
<div id="fig-dotplot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/dotplot-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/dotplot-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenes$ensID, scale =F,
        dot.min = 0, dot.scale = 3, scale.by = &quot;size&quot;) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenes$ensID, labels=selGenes$gene) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/dotplot-3.png" width="364.8" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dotplot-3">
Past versions of dotplot-3.png
</button>
</p>
<div id="fig-dotplot-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/dotplot-3.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/dotplot-3.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dotplot-with-selgenes-label-plus-treatment"
class="section level2">
<h2>Dotplot with selGenes label plus treatment</h2>
<pre class="r"><code>ordDat &lt;- expand.grid(levels(seurat$label), unique(seurat$treatment)) %&gt;% 
  mutate(ord=paste0(Var1, &quot;_&quot;, Var2)) %&gt;% arrange(., Var1)
seurat$label_plus_treat &lt;- factor(seurat$label_plus_treat, levels = ordDat$ord)

Idents(seurat) &lt;- seurat$label_plus_treat

DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenes$ensID, scale =T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenes$ensID, labels=selGenes$gene) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/dotplot%20label%20plus%20treatment-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dotplot-label-plus-treatment-1">
Past versions of "dotplot label plus treatment-1.png"
</button>
</p>
<div id="fig-no-spaces-dotplot-label-plus-treatment-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/dotplot label plus treatment-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="signature-cut-2" class="section level2">
<h2>signature cut 2</h2>
<pre class="r"><code>signDat &lt;- read_delim(file = paste0(basedir,
                    &quot;/data/GSEA/SelSignaturesTreat.txt&quot;),
                    delim = &quot;\t&quot;)
genes &lt;- data.frame(geneID=rownames(seurat)) %&gt;% 
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, geneID))
signDat &lt;- signDat %&gt;% left_join(.,genes, by=&quot;gene&quot;)
allSign &lt;- unique(signDat$signature)

DefaultAssay(object = seurat) &lt;- &quot;integrated&quot;
sce &lt;- as.SingleCellExperiment(seurat)
treatGrps &lt;- unique(sce$treatment)

cutOff &lt;- 2
pal = viridis(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

lapply(unique(signDat$signature), function(sign){
  signGenes &lt;- signDat %&gt;% dplyr::filter(signature == sign)
  sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts)))/nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
  sceSub$sign[which(sceSub$sign &lt; 0)] &lt;- 0
  lapply(treatGrps, function(treat){
    sceSubT &lt;- sceSub[, which(sceSub$treatment == treat)]
    p &lt;- visGroup_adapt(sceSubT, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature - &#39;, treat)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
    p
  })
})</code></pre>
<pre><code>[[1]]
[[1]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-1">
Past versions of "signature cut 2-1.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[1]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-2">
Past versions of "signature cut 2-2.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[2]]
[[2]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-3">
Past versions of "signature cut 2-3.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-3.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-3.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-4">
Past versions of "signature cut 2-4.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-4.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-4.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[3]]
[[3]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-5">
Past versions of "signature cut 2-5.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-5.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-5.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-6">
Past versions of "signature cut 2-6.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-6.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-6.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[4]]
[[4]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-7">
Past versions of "signature cut 2-7.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-7.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-7.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-8">
Past versions of "signature cut 2-8.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-8.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-8.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[5]]
[[5]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-9">
Past versions of "signature cut 2-9.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-9.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-9.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%202-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-2-10">
Past versions of "signature cut 2-10.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-2-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 2-10.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 2-10.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="signature-cut-3" class="section level2">
<h2>signature cut 3</h2>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;integrated&quot;
sce &lt;- as.SingleCellExperiment(seurat)
treatGrps &lt;- unique(sce$treatment)

cutOff &lt;- 3
pal = viridis(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

lapply(unique(signDat$signature), function(sign){
  signGenes &lt;- signDat %&gt;% dplyr::filter(signature == sign)
  sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts)))/nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
  sceSub$sign[which(sceSub$sign &lt; 0)] &lt;- 0
  lapply(treatGrps, function(treat){
    sceSubT &lt;- sceSub[, which(sceSub$treatment == treat)]
    p &lt;- visGroup_adapt(sceSubT, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature - &#39;, treat)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
    p
  })
})</code></pre>
<pre><code>[[1]]
[[1]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-1">
Past versions of "signature cut 3-1.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[1]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-2">
Past versions of "signature cut 3-2.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[2]]
[[2]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-3">
Past versions of "signature cut 3-3.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-3.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-3.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-4">
Past versions of "signature cut 3-4.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-4.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-4.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[3]]
[[3]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-5">
Past versions of "signature cut 3-5.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-5.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-5.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-6">
Past versions of "signature cut 3-6.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-6.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-6.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[4]]
[[4]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-7">
Past versions of "signature cut 3-7.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-7.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-7.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-8">
Past versions of "signature cut 3-8.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-8.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-8.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[5]]
[[5]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-9">
Past versions of "signature cut 3-9.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-9.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-9.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%203-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-3-10">
Past versions of "signature cut 3-10.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-3-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 3-10.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 3-10.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="signature-cut-1.5" class="section level2">
<h2>signature cut 1.5</h2>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;integrated&quot;
sce &lt;- as.SingleCellExperiment(seurat)
treatGrps &lt;- unique(sce$treatment)

cutOff &lt;- 1.5
pal = viridis(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

lapply(unique(signDat$signature), function(sign){
  signGenes &lt;- signDat %&gt;% dplyr::filter(signature == sign)
  sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts)))/nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
  sceSub$sign[which(sceSub$sign &lt; 0)] &lt;- 0
  lapply(treatGrps, function(treat){
    sceSubT &lt;- sceSub[, which(sceSub$treatment == treat)]
    p &lt;- visGroup_adapt(sceSubT, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature - &#39;, treat)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
    p
  })
})</code></pre>
<pre><code>[[1]]
[[1]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-1">
Past versions of "signature cut 1.5-1.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[1]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-2">
Past versions of "signature cut 1.5-2.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[2]]
[[2]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-3">
Past versions of "signature cut 1.5-3.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-3.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-3.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-4">
Past versions of "signature cut 1.5-4.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-4.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-4.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[3]]
[[3]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-5">
Past versions of "signature cut 1.5-5.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-5.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-5.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-6">
Past versions of "signature cut 1.5-6.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-6.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-6.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[4]]
[[4]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-7">
Past versions of "signature cut 1.5-7.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-7.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-7.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-8">
Past versions of "signature cut 1.5-8.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-8.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-8.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[5]]
[[5]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-9">
Past versions of "signature cut 1.5-9.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-9.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-9.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201.5-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-10">
Past versions of "signature cut 1.5-10.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-10.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1.5-10.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="signature-cut-1" class="section level2">
<h2>signature cut 1</h2>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;integrated&quot;
sce &lt;- as.SingleCellExperiment(seurat)
treatGrps &lt;- unique(sce$treatment)

cutOff &lt;- 1
pal = viridis(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

lapply(unique(signDat$signature), function(sign){
  signGenes &lt;- signDat %&gt;% dplyr::filter(signature == sign)
  sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts)))/nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
  sceSub$sign[which(sceSub$sign &lt; 0)] &lt;- 0
  lapply(treatGrps, function(treat){
    sceSubT &lt;- sceSub[, which(sceSub$treatment == treat)]
    p &lt;- visGroup_adapt(sceSubT, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature - &#39;, treat)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
    p
  })
})</code></pre>
<pre><code>[[1]]
[[1]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-1">
Past versions of "signature cut 1-1.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[1]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-2">
Past versions of "signature cut 1-2.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[2]]
[[2]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-3">
Past versions of "signature cut 1-3.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-3.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-3.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-4">
Past versions of "signature cut 1-4.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-4.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-4.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[3]]
[[3]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-5">
Past versions of "signature cut 1-5.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-5.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-5.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-6">
Past versions of "signature cut 1-6.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-6.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-6.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[4]]
[[4]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-7">
Past versions of "signature cut 1-7.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-7.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-7.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-8">
Past versions of "signature cut 1-8.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-8.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-8.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[5]]
[[5]][[1]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-9">
Past versions of "signature cut 1-9.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-9.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-9.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]][[2]]</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/signature%20cut%201-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1-10">
Past versions of "signature cut 1-10.png"
</button>
</p>
<div id="fig-no-spaces-signature-cut-1-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/signature cut 1-10.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/signature cut 1-10.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DefaultAssay(object = seurat) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="vis-fibro-macro-genes" class="section level2">
<h2>vis Fibro + Macro genes</h2>
<pre class="r"><code>selGenes &lt;- read_delim(file = paste0(basedir,
                    &quot;/data/GSEA/SelGenesFibroMacroTreat.txt&quot;),
                    delim = &quot;\t&quot;)
selGenes &lt;- selGenes %&gt;% left_join(.,genes, by=&quot;gene&quot;)
seuratSub &lt;- subset(seurat, label %in% c(&quot;Fibroblasts&quot;, &quot;Macrophages&quot;))

Idents(seuratSub) &lt;- seurat$label_plus_treat
ordDat &lt;- expand.grid(unique(seuratSub$label), unique(seuratSub$treatment)) %&gt;% 
  mutate(ord=paste0(Var1, &quot;_&quot;, Var2)) %&gt;% arrange(., Var1)
gapC &lt;- seq(from=2, to=nrow(ordDat), by=2)
gapR &lt;- NULL

selGenesFibro &lt;- selGenes %&gt;% filter(celltype == &quot;Fibroblast&quot;)
pOut &lt;- avgHeatmap(seurat = seuratSub, selGenes = selGenesFibro,
                  colVecIdent = colLab, 
                  ordVec=ordDat$ord,
                  gapVecR=gapR, gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colTreat)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/plot%20sel%20fibro%20macro%20%20genes-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-sel-fibro-macro--genes-1">
Past versions of "plot sel fibro macro genes-1.png"
</button>
</p>
<div id="fig-no-spaces-plot-sel-fibro-macro--genes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/plot sel fibro macro  genes-1.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/plot sel fibro macro  genes-1.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>selGenesM &lt;- selGenes %&gt;% filter(celltype == &quot;Macrophages&quot;)
pOut &lt;- avgHeatmap(seurat = seuratSub, selGenes = selGenesM,
                  colVecIdent = colLab, 
                  ordVec=ordDat$ord,
                  gapVecR=gapR, gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colTreat)</code></pre>
<p><img src="figure/DEgenesTreatment.Rmd/plot%20sel%20fibro%20macro%20%20genes-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-sel-fibro-macro--genes-2">
Past versions of "plot sel fibro macro genes-2.png"
</button>
</p>
<div id="fig-no-spaces-plot-sel-fibro-macro--genes-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/5c506594387375bcff645a7583c36704005b0a31/docs/figure/DEgenesTreatment.Rmd/plot sel fibro macro  genes-2.png" target="_blank">5c50659</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bf6c6603c3ffa0297dccd4e87e982a6753a3a4cf/docs/figure/DEgenesTreatment.Rmd/plot sel fibro macro  genes-2.png" target="_blank">bf6c660</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-gene-lists" class="section level2">
<h2>save gene lists</h2>
<pre class="r"><code>write.table(overallDE,
            file=paste0(basedir, &quot;/data/MuHeart_Treated_&quot;, 
                        &quot;integratedLabeled_overallDE.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)

write.table(clustDE_Dat,
            file=paste0(basedir,
                        &quot;/data/MuHeart_Treated_&quot;,
                        &quot;integratedLabeled_cwDEGenes.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)

saveRDS(seurat, file=paste0(basedir, &quot;/data/MuHeart_TreatedOnly_&quot;, 
                        &quot;integratedLabeled_seurat.rds&quot;))</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.2               viridisLite_0.4.0          
 [3] enrichplot_1.16.1           DOSE_3.22.0                
 [5] org.Mm.eg.db_3.15.0         AnnotationDbi_1.58.0       
 [7] clusterProfiler_4.4.4       ggpubr_0.4.0               
 [9] pheatmap_1.0.12             ggsci_2.9                  
[11] runSeurat3_0.1.0            here_1.0.1                 
[13] magrittr_2.0.3              sp_1.5-0                   
[15] SeuratObject_4.1.0          Seurat_4.1.1               
[17] forcats_0.5.1               stringr_1.4.0              
[19] dplyr_1.0.9                 purrr_0.3.4                
[21] readr_2.1.2                 tidyr_1.2.0                
[23] tibble_3.1.8                ggplot2_3.3.6              
[25] tidyverse_1.3.2             SingleCellExperiment_1.18.0
[27] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[29] GenomicRanges_1.48.0        GenomeInfoDb_1.32.2        
[31] IRanges_2.30.0              S4Vectors_0.34.0           
[33] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[35] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] scattermore_0.8        bit64_4.0.5            knitr_1.39            
  [4] irlba_2.3.5            DelayedArray_0.22.0    data.table_1.14.2     
  [7] rpart_4.1.16           KEGGREST_1.36.3        RCurl_1.98-1.8        
 [10] generics_0.1.3         cowplot_1.1.1          RSQLite_2.2.15        
 [13] shadowtext_0.1.2       RANN_2.6.1             future_1.27.0         
 [16] bit_4.0.4              tzdb_0.3.0             spatstat.data_2.2-0   
 [19] xml2_1.3.3             lubridate_1.8.0        httpuv_1.6.5          
 [22] assertthat_0.2.1       gargle_1.2.0           xfun_0.31             
 [25] hms_1.1.1              jquerylib_0.1.4        evaluate_0.15         
 [28] promises_1.2.0.1       fansi_1.0.3            dbplyr_2.2.1          
 [31] readxl_1.4.0           igraph_1.3.4           DBI_1.1.3             
 [34] htmlwidgets_1.5.4      spatstat.geom_2.4-0    googledrive_2.0.0     
 [37] ellipsis_0.3.2         backports_1.4.1        deldir_1.0-6          
 [40] vctrs_0.4.1            ROCR_1.0-11            abind_1.4-5           
 [43] cachem_1.0.6           withr_2.5.0            ggforce_0.3.3         
 [46] progressr_0.10.1       vroom_1.5.7            sctransform_0.3.3     
 [49] treeio_1.20.1          goftest_1.2-3          cluster_2.1.3         
 [52] ape_5.6-2              lazyeval_0.2.2         crayon_1.5.1          
 [55] labeling_0.4.2         pkgconfig_2.0.3        tweenr_1.0.2          
 [58] nlme_3.1-158           rlang_1.0.4            globals_0.15.1        
 [61] lifecycle_1.0.1        miniUI_0.1.1.1         downloader_0.4        
 [64] modelr_0.1.8           cellranger_1.1.0       rprojroot_2.0.3       
 [67] polyclip_1.10-0        lmtest_0.9-40          Matrix_1.4-1          
 [70] aplot_0.1.6            carData_3.0-5          zoo_1.8-10            
 [73] reprex_2.0.1           whisker_0.4            ggridges_0.5.3        
 [76] googlesheets4_1.0.0    png_0.1-7              bitops_1.0-7          
 [79] KernSmooth_2.23-20     Biostrings_2.64.0      blob_1.2.3            
 [82] workflowr_1.7.0        qvalue_2.28.0          parallelly_1.32.1     
 [85] spatstat.random_2.2-0  rstatix_0.7.0          gridGraphics_0.5-1    
 [88] ggsignif_0.6.3         scales_1.2.0           memoise_2.0.1         
 [91] plyr_1.8.7             ica_1.0-3              zlibbioc_1.42.0       
 [94] scatterpie_0.1.7       compiler_4.2.1         RColorBrewer_1.1-3    
 [97] fitdistrplus_1.1-8     cli_3.3.0              XVector_0.36.0        
[100] listenv_0.8.0          patchwork_1.1.1        pbapply_1.5-0         
[103] MASS_7.3-58.1          mgcv_1.8-40            tidyselect_1.1.2      
[106] stringi_1.7.8          highr_0.9              yaml_2.3.5            
[109] GOSemSim_2.22.0        ggrepel_0.9.1          grid_4.2.1            
[112] sass_0.4.2             fastmatch_1.1-3        tools_4.2.1           
[115] future.apply_1.9.0     parallel_4.2.1         rstudioapi_0.13       
[118] git2r_0.30.1           gridExtra_2.3          farver_2.1.1          
[121] Rtsne_0.16             ggraph_2.0.5           digest_0.6.29         
[124] rgeos_0.5-9            shiny_1.7.2            Rcpp_1.0.9            
[127] car_3.1-0              broom_1.0.0            later_1.3.0           
[130] RcppAnnoy_0.0.19       httr_1.4.3             colorspace_2.0-3      
[133] rvest_1.0.2            fs_1.5.2               tensor_1.5            
[136] reticulate_1.25        splines_4.2.1          uwot_0.1.11           
[139] yulab.utils_0.0.5      tidytree_0.3.9         spatstat.utils_2.3-1  
[142] graphlayouts_0.8.0     ggplotify_0.1.0        plotly_4.10.0         
[145] xtable_1.8-4           jsonlite_1.8.0         ggtree_3.4.1          
[148] tidygraph_1.2.1        ggfun_0.0.6            R6_2.5.1              
[151] pillar_1.8.0           htmltools_0.5.3        mime_0.12             
[154] glue_1.6.2             fastmap_1.1.0          BiocParallel_1.30.3   
[157] codetools_0.2-18       fgsea_1.22.0           utf8_1.2.2            
[160] lattice_0.20-45        bslib_0.4.0            spatstat.sparse_2.1-1 
[163] leiden_0.4.2           GO.db_3.15.0           survival_3.3-1        
[166] rmarkdown_2.14         munsell_0.5.0          DO.db_2.9             
[169] GenomeInfoDbData_1.2.8 haven_2.5.0            reshape2_1.4.4        
[172] gtable_0.3.0           spatstat.core_2.4-4   </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon Aug 15 17:56:24 2022&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.2               viridisLite_0.4.0          
 [3] enrichplot_1.16.1           DOSE_3.22.0                
 [5] org.Mm.eg.db_3.15.0         AnnotationDbi_1.58.0       
 [7] clusterProfiler_4.4.4       ggpubr_0.4.0               
 [9] pheatmap_1.0.12             ggsci_2.9                  
[11] runSeurat3_0.1.0            here_1.0.1                 
[13] magrittr_2.0.3              sp_1.5-0                   
[15] SeuratObject_4.1.0          Seurat_4.1.1               
[17] forcats_0.5.1               stringr_1.4.0              
[19] dplyr_1.0.9                 purrr_0.3.4                
[21] readr_2.1.2                 tidyr_1.2.0                
[23] tibble_3.1.8                ggplot2_3.3.6              
[25] tidyverse_1.3.2             SingleCellExperiment_1.18.0
[27] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[29] GenomicRanges_1.48.0        GenomeInfoDb_1.32.2        
[31] IRanges_2.30.0              S4Vectors_0.34.0           
[33] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[35] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] scattermore_0.8        bit64_4.0.5            knitr_1.39            
  [4] irlba_2.3.5            DelayedArray_0.22.0    data.table_1.14.2     
  [7] rpart_4.1.16           KEGGREST_1.36.3        RCurl_1.98-1.8        
 [10] generics_0.1.3         cowplot_1.1.1          RSQLite_2.2.15        
 [13] shadowtext_0.1.2       RANN_2.6.1             future_1.27.0         
 [16] bit_4.0.4              tzdb_0.3.0             spatstat.data_2.2-0   
 [19] xml2_1.3.3             lubridate_1.8.0        httpuv_1.6.5          
 [22] assertthat_0.2.1       gargle_1.2.0           xfun_0.31             
 [25] hms_1.1.1              jquerylib_0.1.4        evaluate_0.15         
 [28] promises_1.2.0.1       fansi_1.0.3            dbplyr_2.2.1          
 [31] readxl_1.4.0           igraph_1.3.4           DBI_1.1.3             
 [34] htmlwidgets_1.5.4      spatstat.geom_2.4-0    googledrive_2.0.0     
 [37] ellipsis_0.3.2         backports_1.4.1        deldir_1.0-6          
 [40] vctrs_0.4.1            ROCR_1.0-11            abind_1.4-5           
 [43] cachem_1.0.6           withr_2.5.0            ggforce_0.3.3         
 [46] progressr_0.10.1       vroom_1.5.7            sctransform_0.3.3     
 [49] treeio_1.20.1          goftest_1.2-3          cluster_2.1.3         
 [52] ape_5.6-2              lazyeval_0.2.2         crayon_1.5.1          
 [55] labeling_0.4.2         pkgconfig_2.0.3        tweenr_1.0.2          
 [58] nlme_3.1-158           rlang_1.0.4            globals_0.15.1        
 [61] lifecycle_1.0.1        miniUI_0.1.1.1         downloader_0.4        
 [64] modelr_0.1.8           cellranger_1.1.0       rprojroot_2.0.3       
 [67] polyclip_1.10-0        lmtest_0.9-40          Matrix_1.4-1          
 [70] aplot_0.1.6            carData_3.0-5          zoo_1.8-10            
 [73] reprex_2.0.1           whisker_0.4            ggridges_0.5.3        
 [76] googlesheets4_1.0.0    png_0.1-7              bitops_1.0-7          
 [79] KernSmooth_2.23-20     Biostrings_2.64.0      blob_1.2.3            
 [82] workflowr_1.7.0        qvalue_2.28.0          parallelly_1.32.1     
 [85] spatstat.random_2.2-0  rstatix_0.7.0          gridGraphics_0.5-1    
 [88] ggsignif_0.6.3         scales_1.2.0           memoise_2.0.1         
 [91] plyr_1.8.7             ica_1.0-3              zlibbioc_1.42.0       
 [94] scatterpie_0.1.7       compiler_4.2.1         RColorBrewer_1.1-3    
 [97] fitdistrplus_1.1-8     cli_3.3.0              XVector_0.36.0        
[100] listenv_0.8.0          patchwork_1.1.1        pbapply_1.5-0         
[103] MASS_7.3-58.1          mgcv_1.8-40            tidyselect_1.1.2      
[106] stringi_1.7.8          highr_0.9              yaml_2.3.5            
[109] GOSemSim_2.22.0        ggrepel_0.9.1          grid_4.2.1            
[112] sass_0.4.2             fastmatch_1.1-3        tools_4.2.1           
[115] future.apply_1.9.0     parallel_4.2.1         rstudioapi_0.13       
[118] git2r_0.30.1           gridExtra_2.3          farver_2.1.1          
[121] Rtsne_0.16             ggraph_2.0.5           digest_0.6.29         
[124] rgeos_0.5-9            shiny_1.7.2            Rcpp_1.0.9            
[127] car_3.1-0              broom_1.0.0            later_1.3.0           
[130] RcppAnnoy_0.0.19       httr_1.4.3             colorspace_2.0-3      
[133] rvest_1.0.2            fs_1.5.2               tensor_1.5            
[136] reticulate_1.25        splines_4.2.1          uwot_0.1.11           
[139] yulab.utils_0.0.5      tidytree_0.3.9         spatstat.utils_2.3-1  
[142] graphlayouts_0.8.0     ggplotify_0.1.0        plotly_4.10.0         
[145] xtable_1.8-4           jsonlite_1.8.0         ggtree_3.4.1          
[148] tidygraph_1.2.1        ggfun_0.0.6            R6_2.5.1              
[151] pillar_1.8.0           htmltools_0.5.3        mime_0.12             
[154] glue_1.6.2             fastmap_1.1.0          BiocParallel_1.30.3   
[157] codetools_0.2-18       fgsea_1.22.0           utf8_1.2.2            
[160] lattice_0.20-45        bslib_0.4.0            spatstat.sparse_2.1-1 
[163] leiden_0.4.2           GO.db_3.15.0           survival_3.3-1        
[166] rmarkdown_2.14         munsell_0.5.0          DO.db_2.9             
[169] GenomeInfoDbData_1.2.8 haven_2.5.0            reshape2_1.4.4        
[172] gtable_0.3.0           spatstat.core_2.4-4   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
