<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2021-04-29" />

<title>Merge CD45+ cells</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">TCRM_Christina</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Merge CD45+ cells</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">29 April 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>TCRM_Christina/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200518code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200518)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200518code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200518)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeTCRMChristinatree8eb0b02645b47d8e87ae2a02177fe7d4919436b8targetblank8eb0b02a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/TCRM_Christina/tree/8eb0b02645b47d8e87ae2a02177fe7d4919436b8" target="_blank">8eb0b02</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeTCRMChristinatree8eb0b02645b47d8e87ae2a02177fe7d4919436b8targetblank8eb0b02a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/TCRM_Christina/tree/8eb0b02645b47d8e87ae2a02177fe7d4919436b8" target="_blank">8eb0b02</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/data_old/
    Ignored:    output/.DS_Store

Unstaged changes:
    Modified:   analysis/mergeCD45cells.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/mergeCD45cells.Rmd</code>) and HTML (<code>docs/mergeCD45cells.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/analysis/mergeCD45cells.Rmd" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
<td>
update hematopoietic cells
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/mergeCD45cells.html" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
<td>
update hematopoietic cells
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bfce29b5eb4bc848c196f740bb69bd2eb09e5723/analysis/mergeCD45cells.Rmd" target="_blank">bfce29b</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-10
</td>
<td>
update CD45
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/bfce29b5eb4bc848c196f740bb69bd2eb09e5723/docs/mergeCD45cells.html" target="_blank">bfce29b</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-10
</td>
<td>
update CD45
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/d7e2251c34c1da03ecef9119af1283088378926d/analysis/mergeCD45cells.Rmd" target="_blank">d7e2251</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-04
</td>
<td>
analysis CD45 pos cells
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/TCRM_Christina/d7e2251c34c1da03ecef9119af1283088378926d/docs/mergeCD45cells.html" target="_blank">d7e2251</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-04
</td>
<td>
analysis CD45 pos cells
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(clusterProfiler)
  library(org.Mm.eg.db)
  library(DOSE)
  library(enrichplot)
  library(msigdbr)
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(ggpubr)
  library(pheatmap)
})</code></pre>
</div>
<div id="set-dir" class="section level2">
<h2>set dir</h2>
<pre class="r"><code>basedir &lt;- here()
metaDat &lt;- read_tsv(paste0(basedir, &quot;/metadata_lymphcytes.txt&quot;), col_names = T)</code></pre>
</div>
<div id="load-and-assign-samples" class="section level2">
<h2>load and assign samples</h2>
<pre class="r"><code>assignSamples &lt;- function(smpNam, basedirSmp, smpGrp, smpBatch){
  smpNamFull &lt;- list.files(path = paste0(basedirSmp, &quot;/data/&quot;),
                 pattern = paste0(smpNam, &quot;.*_seurat.rds&quot;))
  seuratSmp &lt;- readRDS(paste0(basedirSmp, &quot;/data/&quot;, smpNamFull))
  seuratSmp$grp &lt;- smpGrp
  seuratSmp$batch &lt;- smpBatch
  return(seuratSmp)
}

####################################################################

for(i in 1:length(metaDat$Sample)){
  seuratX &lt;- assignSamples(smpNam = metaDat$Sample[i],
                           basedirSmp = basedir,
                           smpGrp = metaDat$group[i],
                           smpBatch = metaDat$batch[i])
  if(exists(&quot;seurat&quot;)){
    seurat &lt;- merge(x = seurat, y = seuratX, project = &quot;TCRMxCD45&quot;)
  }else{
    seurat &lt;- seuratX
  }
}

remove(seuratX)</code></pre>
</div>
<div id="remove-contaminating-cells" class="section level2">
<h2>remove contaminating cells</h2>
<pre class="r"><code>seurat &lt;- subset(seurat, subset = ENSMUSG00000001506.Col1a1 == 0)
seurat &lt;- subset(seurat, subset = ENSMUSG00000029231.Pdgfra == 0)
seurat &lt;- subset(seurat, subset = ENSMUSG00000024620.Pdgfrb == 0)
seurat &lt;- subset(seurat, subset = ENSMUSG00000020717.Pecam1 == 0)</code></pre>
</div>
<div id="run-clustering-and-dr-and-remove-contaminating-cells" class="section level2">
<h2>run clustering and DR and remove contaminating cells</h2>
<pre class="r"><code>seurat &lt;- rerunSeurat3(seurat, res=c(0.6, 0.4, 0.25))</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5686
Number of edges: 200984

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9062
Number of communities: 18
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5686
Number of edges: 200984

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9239
Number of communities: 16
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5686
Number of edges: 200984

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9406
Number of communities: 14
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>dim(seurat)</code></pre>
<pre><code>[1] 23866  5686</code></pre>
<pre class="r"><code>seurat$cluster_plus_cond &lt;- paste0(seurat$seurat_clusters, &quot;_&quot;, seurat$grp)


## remove contaminating endothelial cells
seurat &lt;- subset(seurat, idents = &quot;11&quot;, invert=T)
seurat &lt;- RunPCA(object = seurat, npcs = 30, verbose = FALSE)
seurat &lt;- RunTSNE(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)
seurat &lt;- RunUMAP(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)</code></pre>
</div>
<div id="assign-celltypes" class="section level2">
<h2>assign celltypes</h2>
<pre class="r"><code>## based on markers 
seurat$clusterLabel &lt;- &quot;MacrophagesMonocytes&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;2&quot;)] &lt;- &quot;Bcells&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;5&quot;)] &lt;- &quot;Tcells&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;6&quot;)] &lt;- &quot;NKcells&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;7&quot;)] &lt;- &quot;Neutrophils&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;8&quot;)] &lt;- &quot;proliferating&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;9&quot;)] &lt;- &quot;DCs&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;10&quot;)] &lt;- &quot;ILCs&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;12&quot;)] &lt;- &quot;pDCs&quot;
seurat$clusterLabel[which(seurat$seurat_clusters == &quot;13&quot;)] &lt;- &quot;Basophils&quot;

seurat$label_plus_cond &lt;- paste0(seurat$clusterLabel, &quot;_&quot;, seurat$grp)</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- pal_igv()(length(levels(seurat)))
colGrp &lt;- pal_jama()(length(unique(seurat$grp)))
colSmp &lt;- pal_uchicago()(length(unique(seurat$dataset)))
colLab &lt;- pal_futurama()(length(unique(seurat$clusterLabel)))

names(colPal) &lt;- levels(seurat)
names(colGrp) &lt;- unique(seurat$grp)
names(colSmp) &lt;- unique(seurat$dataset)
names(colLab) &lt;- unique(seurat$clusterLabel)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2>vis data</h2>
<div id="clusters" class="section level3">
<h3>clusters</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/vis%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-1">
Past versions of “vis cluster-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/vis cluster-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bfce29b5eb4bc848c196f740bb69bd2eb09e5723/docs/figure/mergeCD45cells.Rmd/vis cluster-1.png" target="_blank">bfce29b</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/d7e2251c34c1da03ecef9119af1283088378926d/docs/figure/mergeCD45cells.Rmd/vis cluster-1.png" target="_blank">d7e2251</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="group" class="section level3">
<h3>group</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;grp&quot;, cols=colGrp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/vis%20grp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-grp-1">
Past versions of “vis grp-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-grp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/vis grp-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bfce29b5eb4bc848c196f740bb69bd2eb09e5723/docs/figure/mergeCD45cells.Rmd/vis grp-1.png" target="_blank">bfce29b</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/d7e2251c34c1da03ecef9119af1283088378926d/docs/figure/mergeCD45cells.Rmd/vis grp-1.png" target="_blank">d7e2251</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;clusterLabel&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/vis%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-1">
Past versions of “vis label-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-label-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/vis label-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cell-counts" class="section level2 tabset">
<h2>cell counts</h2>
<div id="overall-cell-counts" class="section level3">
<h3>overall cell counts</h3>
<pre class="r"><code>dim(seurat)[2]</code></pre>
<pre><code>[1] 5618</code></pre>
</div>
<div id="counts-per-group" class="section level3">
<h3>counts per group</h3>
<pre class="r"><code>knitr::kable(table(seurat$grp))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">LM_CD45</td>
<td align="right">3435</td>
</tr>
<tr class="even">
<td align="left">TCRM_CD45</td>
<td align="right">2183</td>
</tr>
</tbody>
</table>
</div>
<div id="counts-per-label-and-group" class="section level3">
<h3>counts per label and group</h3>
<pre class="r"><code>knitr::kable(table(seurat$label_plus_cond))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Basophils_LM_CD45</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">Basophils_TCRM_CD45</td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">Bcells_LM_CD45</td>
<td align="right">668</td>
</tr>
<tr class="even">
<td align="left">Bcells_TCRM_CD45</td>
<td align="right">199</td>
</tr>
<tr class="odd">
<td align="left">DCs_LM_CD45</td>
<td align="right">63</td>
</tr>
<tr class="even">
<td align="left">DCs_TCRM_CD45</td>
<td align="right">60</td>
</tr>
<tr class="odd">
<td align="left">ILCs_LM_CD45</td>
<td align="right">66</td>
</tr>
<tr class="even">
<td align="left">ILCs_TCRM_CD45</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">MacrophagesMonocytes_LM_CD45</td>
<td align="right">2113</td>
</tr>
<tr class="even">
<td align="left">MacrophagesMonocytes_TCRM_CD45</td>
<td align="right">1188</td>
</tr>
<tr class="odd">
<td align="left">Neutrophils_LM_CD45</td>
<td align="right">141</td>
</tr>
<tr class="even">
<td align="left">Neutrophils_TCRM_CD45</td>
<td align="right">172</td>
</tr>
<tr class="odd">
<td align="left">NKcells_LM_CD45</td>
<td align="right">176</td>
</tr>
<tr class="even">
<td align="left">NKcells_TCRM_CD45</td>
<td align="right">190</td>
</tr>
<tr class="odd">
<td align="left">pDCs_LM_CD45</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">pDCs_TCRM_CD45</td>
<td align="right">39</td>
</tr>
<tr class="odd">
<td align="left">proliferating_LM_CD45</td>
<td align="right">72</td>
</tr>
<tr class="even">
<td align="left">proliferating_TCRM_CD45</td>
<td align="right">57</td>
</tr>
<tr class="odd">
<td align="left">Tcells_LM_CD45</td>
<td align="right">115</td>
</tr>
<tr class="even">
<td align="left">Tcells_TCRM_CD45</td>
<td align="right">252</td>
</tr>
</tbody>
</table>
</div>
<div id="counts-per-cluster-and-group" class="section level3">
<h3>counts per cluster and group</h3>
<pre class="r"><code>knitr::kable(table(seurat$cluster_plus_cond))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0_LM_CD45</td>
<td align="right">1350</td>
</tr>
<tr class="even">
<td align="left">0_TCRM_CD45</td>
<td align="right">55</td>
</tr>
<tr class="odd">
<td align="left">1_LM_CD45</td>
<td align="right">308</td>
</tr>
<tr class="even">
<td align="left">1_TCRM_CD45</td>
<td align="right">650</td>
</tr>
<tr class="odd">
<td align="left">10_LM_CD45</td>
<td align="right">66</td>
</tr>
<tr class="even">
<td align="left">10_TCRM_CD45</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">12_LM_CD45</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">12_TCRM_CD45</td>
<td align="right">39</td>
</tr>
<tr class="odd">
<td align="left">13_LM_CD45</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">13_TCRM_CD45</td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">2_LM_CD45</td>
<td align="right">668</td>
</tr>
<tr class="even">
<td align="left">2_TCRM_CD45</td>
<td align="right">199</td>
</tr>
<tr class="odd">
<td align="left">3_LM_CD45</td>
<td align="right">448</td>
</tr>
<tr class="even">
<td align="left">3_TCRM_CD45</td>
<td align="right">47</td>
</tr>
<tr class="odd">
<td align="left">4_LM_CD45</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">4_TCRM_CD45</td>
<td align="right">436</td>
</tr>
<tr class="odd">
<td align="left">5_LM_CD45</td>
<td align="right">115</td>
</tr>
<tr class="even">
<td align="left">5_TCRM_CD45</td>
<td align="right">252</td>
</tr>
<tr class="odd">
<td align="left">6_LM_CD45</td>
<td align="right">176</td>
</tr>
<tr class="even">
<td align="left">6_TCRM_CD45</td>
<td align="right">190</td>
</tr>
<tr class="odd">
<td align="left">7_LM_CD45</td>
<td align="right">141</td>
</tr>
<tr class="even">
<td align="left">7_TCRM_CD45</td>
<td align="right">172</td>
</tr>
<tr class="odd">
<td align="left">8_LM_CD45</td>
<td align="right">72</td>
</tr>
<tr class="even">
<td align="left">8_TCRM_CD45</td>
<td align="right">57</td>
</tr>
<tr class="odd">
<td align="left">9_LM_CD45</td>
<td align="right">63</td>
</tr>
<tr class="even">
<td align="left">9_TCRM_CD45</td>
<td align="right">60</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="marker-genes" class="section level2">
<h2>marker genes</h2>
<pre class="r"><code>seurat_markers_all &lt;- FindAllMarkers(object = seurat, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.25,
                                     logfc.threshold = 0.25,
                                     test.use = &quot;wilcox&quot;)</code></pre>
</div>
<div id="vis-cluster-characterization" class="section level2">
<h2>vis cluster characterization</h2>
<pre class="r"><code>selMarker &lt;- read_tsv(paste0(basedir, &quot;/docs/markerGenesHemato.txt&quot;))
genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))
selMarkerAll &lt;- selMarker %&gt;% left_join(., genes, by=&quot;geneID&quot;)

grpCnt &lt;- selMarker %&gt;% group_by(Cluster) %&gt;% summarise(cnt=n())
gapR &lt;- data.frame(Cluster=unique(selMarker$Cluster)) %&gt;% 
  left_join(.,grpCnt, by=&quot;Cluster&quot;) %&gt;% mutate(cumSum=cumsum(cnt)) 

Idents(seurat) &lt;- seurat$seurat_clusters
avgHeatmap(seurat = seurat, selGenes = selMarkerAll,
                  colVecIdent = colPal, 
                  ordVec=levels(seurat)[order(levels(seurat))],
                  gapVecR=gapR$cumSum, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/vis%20cluster%20characteization-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-characteization-1">
Past versions of “vis cluster characteization-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-characteization-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/vis cluster characteization-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterLabel
avgHeatmap(seurat = seurat, selGenes = selMarkerAll,
                  colVecIdent = colLab, 
                  ordVec=levels(seurat)[order(levels(seurat))],
                  gapVecR=gapR$cumSum, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/vis%20cluster%20characteization-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-characteization-2">
Past versions of “vis cluster characteization-2.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-characteization-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/vis cluster characteization-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seurat) &lt;- seurat$label_plus_cond
avgHeatmap(seurat = seurat, selGenes = selMarkerAll,
                  colVecIdent = colLab, colVecCond = colGrp, 
                  ordVec=levels(seurat)[order(levels(seurat))],
                  gapVecR=gapR$cumSum, gapVecC=NULL,cc=T,
                  cr=F, condCol=T)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/vis%20cluster%20characteization-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-characteization-3">
Past versions of “vis cluster characteization-3.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-characteization-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/vis cluster characteization-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>avgHeatmap(seurat = seurat, selGenes = selMarkerAll,
                  colVecIdent = colLab, colVecCond = colGrp, 
                  ordVec=levels(seurat)[order(levels(seurat))],
                  gapVecR=gapR$cumSum, gapVecC=NULL,cc=F,
                  cr=F, condCol=T)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/vis%20cluster%20characteization-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-characteization-4">
Past versions of “vis cluster characteization-4.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-characteization-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/vis cluster characteization-4.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="overall-de-genes" class="section level2">
<h2>overall DE genes</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$grp

DEgenes &lt;- FindAllMarkers(object = seurat, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.25,
                                     logfc.threshold = 0.25,
                                     test.use = &quot;wilcox&quot;)

### plot DE genes top 15 avg logFC
DEGenesAll &lt;- DEgenes %&gt;% group_by(cluster) %&gt;% 
  top_n(15, avg_log2FC)

Idents(seurat) &lt;- seurat$cluster_plus_cond
avgHeatmap(seurat = seurat, selGenes = DEGenesAll,
                  colVecIdent = colPal, colVecCond = colGrp, 
                  ordVec=levels(seurat)[order(levels(seurat))],
                  gapVecR=NULL, gapVecC=NULL,cc=F,
                  cr=T, condCol=T)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/overall%20DE-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-overall-DE-1">
Past versions of “overall DE-1.png”
</button>
</p>
<div id="fig-no-spaces-overall-DE-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/overall DE-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/bfce29b5eb4bc848c196f740bb69bd2eb09e5723/docs/figure/mergeCD45cells.Rmd/overall DE-1.png" target="_blank">bfce29b</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/d7e2251c34c1da03ecef9119af1283088378926d/docs/figure/mergeCD45cells.Rmd/overall DE-1.png" target="_blank">d7e2251</a>
</td>
<td>
mluetge
</td>
<td>
2020-11-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seurat) &lt;- seurat$label_plus_cond
avgHeatmap(seurat = seurat, selGenes = DEGenesAll,
                  colVecIdent = colLab, colVecCond = colGrp, 
                  ordVec=levels(seurat)[order(levels(seurat))],
                  gapVecR=NULL, gapVecC=NULL,cc=F,
                  cr=T, condCol=T)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/overall%20DE-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-overall-DE-2">
Past versions of “overall DE-2.png”
</button>
</p>
<div id="fig-no-spaces-overall-DE-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/overall DE-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="clusterwise-de-genes" class="section level2">
<h2>clusterwise DE genes</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$grp
seurat$clusterLabel &lt;- as.factor(seurat$clusterLabel)
grpVec &lt;- levels(seurat$clusterLabel)

clustDE &lt;- lapply(grpVec, function(grp){
  grpSub &lt;- levels(seurat$clusterLabel)[which(levels(seurat$clusterLabel)==grp)]
  seuratSub &lt;- subset(seurat, clusterLabel == grpSub)
  DEgenes &lt;-FindAllMarkers(seuratSub, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.1)
  DEgenes &lt;- DEgenes %&gt;% filter(p_val_adj&lt;0.01) %&gt;%
    mutate(group=paste0(grp, &quot;_&quot;, cluster))
})

names(clustDE) &lt;- grpVec

clustDE_Dat &lt;- data.frame(do.call(&quot;rbind&quot;, clustDE))</code></pre>
<div id="avg-heatmap-top-cwde-genes" class="section level3 tabset">
<h3>avg Heatmap top cwDE genes</h3>
<div id="bcells" class="section level4">
<h4>Bcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE Bcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Bcells-1">
Past versions of “avgHeat cwDE Bcells-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Bcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE Bcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dcs" class="section level4">
<h4>DCs</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE DCs-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-DCs-1">
Past versions of “avgHeat cwDE DCs-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-DCs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE DCs-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ilcs" class="section level4">
<h4>ILCs</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE ILCs-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-ILCs-1">
Past versions of “avgHeat cwDE ILCs-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-ILCs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE ILCs-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="macrophagesmonocytes" class="section level4">
<h4>MacrophagesMonocytes</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE MacrophagesMonocytes-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-MacrophagesMonocytes-1">
Past versions of “avgHeat cwDE MacrophagesMonocytes-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-MacrophagesMonocytes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE MacrophagesMonocytes-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="neutrophils" class="section level4">
<h4>Neutrophils</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE Neutrophils-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Neutrophils-1">
Past versions of “avgHeat cwDE Neutrophils-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Neutrophils-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE Neutrophils-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nkcells" class="section level4">
<h4>NKcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE NKcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-NKcells-1">
Past versions of “avgHeat cwDE NKcells-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-NKcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE NKcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proliferating" class="section level4">
<h4>proliferating</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE proliferating-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-proliferating-1">
Past versions of “avgHeat cwDE proliferating-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-proliferating-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE proliferating-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tcells" class="section level4">
<h4>Tcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/avgHeat cwDE Tcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avgHeat-cwDE-Tcells-1">
Past versions of “avgHeat cwDE Tcells-1.png”
</button>
</p>
<div id="fig-no-spaces-avgHeat-cwDE-Tcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/avgHeat cwDE Tcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="save-objects" class="section level2">
<h2>save objects</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$seurat_clusters
saveRDS(seurat, file = paste0(basedir, 
                              &quot;/data/Mu_Heart_TCRMplusLM_CD45_seurat.rds&quot;))

write.table(seurat_markers_all,
            file=paste0(basedir, &quot;/data/Mu_Heart_TCRMplusLM_CD45_markerGenes.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)

write.table(DEgenes,
            file=paste0(basedir,
                        &quot;/data/Mu_Heart_TCRMplusLM_CD45_overall_DEGenes.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)

write.table(clustDE_Dat,
            file=paste0(basedir,
                        &quot;/data/Mu_Heart_TCRMplusLM_CD45_cwDEGenes.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="gsea-clusterprofiler-overall-de-genes" class="section level2 tabset">
<h2>GSEA clusterProfiler overall DE genes</h2>
<div id="overall-de-genes-lm_cd45-enriched" class="section level3">
<h3>overall DE genes LM_CD45 enriched</h3>
<pre class="r"><code>DEgenesCond &lt;- DEgenes %&gt;% filter(cluster == &quot;LM_CD45&quot;) %&gt;%
  mutate(EnsID = gsub(&quot;\\..*$&quot;, &quot;&quot;, gene))

egoLM &lt;- enrichGO(gene         = DEgenesCond$EnsID,
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
egoLM &lt;- setReadable(egoLM, OrgDb = org.Mm.eg.db)
dotplot(egoLM, showCategory=30)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/GSEA%20clusterprofiler%20LM_CD45-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-LM_CD45-1">
Past versions of “GSEA clusterprofiler LM_CD45-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-LM_CD45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/GSEA clusterprofiler LM_CD45-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(egoLM, node_label=&quot;all&quot;) </code></pre>
<p><img src="figure/mergeCD45cells.Rmd/GSEA%20clusterprofiler%20LM_CD45-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-LM_CD45-2">
Past versions of “GSEA clusterprofiler LM_CD45-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-LM_CD45-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/GSEA clusterprofiler LM_CD45-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>emapplot(egoLM, showCategory = 15)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/GSEA%20clusterprofiler%20LM_CD45-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-LM_CD45-3">
Past versions of “GSEA clusterprofiler LM_CD45-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-LM_CD45-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/GSEA clusterprofiler LM_CD45-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="overall-de-genes-tcrm_cd45-enriched" class="section level3">
<h3>overall DE genes TCRM_CD45 enriched</h3>
<pre class="r"><code>DEgenesCond &lt;- DEgenes %&gt;% filter(cluster == &quot;TCRM_CD45&quot;) %&gt;%
  mutate(EnsID = gsub(&quot;\\..*$&quot;, &quot;&quot;, gene))

egoLM &lt;- enrichGO(gene         = DEgenesCond$EnsID,
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
egoLM &lt;- setReadable(egoLM, OrgDb = org.Mm.eg.db)
dotplot(egoLM, showCategory=30)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/GSEA%20clusterprofiler%20TCRM_CD45-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-TCRM_CD45-1">
Past versions of “GSEA clusterprofiler TCRM_CD45-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-TCRM_CD45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/GSEA clusterprofiler TCRM_CD45-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(egoLM, node_label=&quot;all&quot;) </code></pre>
<p><img src="figure/mergeCD45cells.Rmd/GSEA%20clusterprofiler%20TCRM_CD45-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-TCRM_CD45-2">
Past versions of “GSEA clusterprofiler TCRM_CD45-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-TCRM_CD45-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/GSEA clusterprofiler TCRM_CD45-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>emapplot(egoLM, showCategory = 15)</code></pre>
<p><img src="figure/mergeCD45cells.Rmd/GSEA%20clusterprofiler%20TCRM_CD45-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-TCRM_CD45-3">
Past versions of “GSEA clusterprofiler TCRM_CD45-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-TCRM_CD45-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/docs/figure/mergeCD45cells.Rmd/GSEA clusterprofiler TCRM_CD45-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="gsea-clusterprofiler-clusterwise-de-genes" class="section level2 tabset">
<h2>GSEA clusterProfiler clusterwise DE genes</h2>
<div id="lm_cd45-enriched" class="section level3 tabset">
<h3>LM_CD45 enriched</h3>
<pre class="r"><code>clustDE_DatGSEA &lt;- clustDE_Dat %&gt;% 
  mutate(clusterLabel=gsub(&quot;_.*$&quot;, &quot;&quot;, group)) %&gt;% 
  filter(cluster == &quot;LM_CD45&quot;) %&gt;%
  mutate(EnsID = gsub(&quot;\\..*$&quot;, &quot;&quot;, gene))


grpVec &lt;- as.character(unique(clustDE_DatGSEA$clusterLabel))

template_ep &lt;- c(
    &quot;#### {{grp}}\n&quot;,
    &quot;```{r GSEA cwDE LM_CD45 {{grp}}, echo=FALSE,fig.height=10,fig.width=10}\n&quot;,
    &quot;DeSub &lt;- clustDE_DatGSEA[which(
                    clustDE_DatGSEA$clusterLabel == as.character(&#39;{{grp}}&#39;)),]&quot;,
    &quot;egoLM &lt;- enrichGO(gene   = DeSub$EnsID,
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &#39;BP&#39;,
                pAdjustMethod = &#39;BH&#39;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)&quot;,
    &quot;if(!is.null(egoLM)){
      egoLM &lt;- setReadable(egoLM, OrgDb = org.Mm.eg.db)
      dotplot(egoLM, showCategory=30)
    }&quot;,
    &quot;if(!is.null(egoLM)){
      cnetplot(egoLM, node_label=&#39;all&#39;)
    }&quot;,
    &quot;if(!is.null(egoLM)){
      emapplot(egoLM, showCategory = 15)
    }\n&quot;,
    &quot;```\n&quot;,
    &quot;\n&quot;
)
    
plots_ep &lt;- lapply(grpVec, 
  function(grp) knitr::knit_expand(text = template_ep)
)</code></pre>
<div id="bcells-1" class="section level4">
<h4>Bcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Bcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Bcells-1">
Past versions of “GSEA cwDE LM_CD45 Bcells-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Bcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Bcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Bcells-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Bcells-2">
Past versions of “GSEA cwDE LM_CD45 Bcells-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Bcells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Bcells-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Bcells-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Bcells-3">
Past versions of “GSEA cwDE LM_CD45 Bcells-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Bcells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Bcells-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dcs-1" class="section level4">
<h4>DCs</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 DCs-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-DCs-1">
Past versions of “GSEA cwDE LM_CD45 DCs-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-DCs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 DCs-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 DCs-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-DCs-2">
Past versions of “GSEA cwDE LM_CD45 DCs-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-DCs-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 DCs-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 DCs-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-DCs-3">
Past versions of “GSEA cwDE LM_CD45 DCs-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-DCs-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 DCs-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="macrophagesmonocytes-1" class="section level4">
<h4>MacrophagesMonocytes</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 MacrophagesMonocytes-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-MacrophagesMonocytes-1">
Past versions of “GSEA cwDE LM_CD45 MacrophagesMonocytes-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-MacrophagesMonocytes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 MacrophagesMonocytes-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 MacrophagesMonocytes-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-MacrophagesMonocytes-2">
Past versions of “GSEA cwDE LM_CD45 MacrophagesMonocytes-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-MacrophagesMonocytes-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 MacrophagesMonocytes-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 MacrophagesMonocytes-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-MacrophagesMonocytes-3">
Past versions of “GSEA cwDE LM_CD45 MacrophagesMonocytes-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-MacrophagesMonocytes-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 MacrophagesMonocytes-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="neutrophils-1" class="section level4">
<h4>Neutrophils</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Neutrophils-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Neutrophils-1">
Past versions of “GSEA cwDE LM_CD45 Neutrophils-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Neutrophils-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Neutrophils-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Neutrophils-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Neutrophils-2">
Past versions of “GSEA cwDE LM_CD45 Neutrophils-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Neutrophils-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Neutrophils-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Neutrophils-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Neutrophils-3">
Past versions of “GSEA cwDE LM_CD45 Neutrophils-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Neutrophils-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Neutrophils-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nkcells-1" class="section level4">
<h4>NKcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 NKcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-NKcells-1">
Past versions of “GSEA cwDE LM_CD45 NKcells-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-NKcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 NKcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 NKcells-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-NKcells-2">
Past versions of “GSEA cwDE LM_CD45 NKcells-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-NKcells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 NKcells-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 NKcells-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-NKcells-3">
Past versions of “GSEA cwDE LM_CD45 NKcells-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-NKcells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 NKcells-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pdcs" class="section level4">
<h4>pDCs</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 pDCs-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-pDCs-1">
Past versions of “GSEA cwDE LM_CD45 pDCs-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-pDCs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 pDCs-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 pDCs-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-pDCs-2">
Past versions of “GSEA cwDE LM_CD45 pDCs-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-pDCs-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 pDCs-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 pDCs-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-pDCs-3">
Past versions of “GSEA cwDE LM_CD45 pDCs-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-pDCs-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 pDCs-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proliferating-1" class="section level4">
<h4>proliferating</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 proliferating-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-proliferating-1">
Past versions of “GSEA cwDE LM_CD45 proliferating-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-proliferating-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 proliferating-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 proliferating-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-proliferating-2">
Past versions of “GSEA cwDE LM_CD45 proliferating-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-proliferating-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 proliferating-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 proliferating-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-proliferating-3">
Past versions of “GSEA cwDE LM_CD45 proliferating-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-proliferating-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 proliferating-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tcells-1" class="section level4">
<h4>Tcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Tcells-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Tcells-1">
Past versions of “GSEA cwDE LM_CD45 Tcells-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Tcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Tcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Tcells-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Tcells-2">
Past versions of “GSEA cwDE LM_CD45 Tcells-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Tcells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Tcells-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Tcells-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-LM_CD45-Tcells-3">
Past versions of “GSEA cwDE LM_CD45 Tcells-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-LM_CD45-Tcells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE LM_CD45 Tcells-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="tcrm_cd45-enriched" class="section level3 tabset">
<h3>TCRM_CD45 enriched</h3>
<pre class="r"><code>clustDE_DatGSEA &lt;- clustDE_Dat %&gt;% 
  mutate(clusterLabel=gsub(&quot;_.*$&quot;, &quot;&quot;, group)) %&gt;% 
  filter(cluster == &quot;TCRM_CD45&quot;) %&gt;%
  mutate(EnsID = gsub(&quot;\\..*$&quot;, &quot;&quot;, gene))


grpVec &lt;- as.character(unique(clustDE_DatGSEA$clusterLabel))

template_ep &lt;- c(
    &quot;#### {{grp}}\n&quot;,
    &quot;```{r GSEA cwDE TCRM_CD45 {{grp}},echo=FALSE,fig.height=10,fig.width=18}\n&quot;,
    &quot;DeSub &lt;- clustDE_DatGSEA[which(
                    clustDE_DatGSEA$clusterLabel == as.character(&#39;{{grp}}&#39;)),]&quot;,
    &quot;egoLM &lt;- enrichGO(gene   = DeSub$EnsID,
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &#39;BP&#39;,
                pAdjustMethod = &#39;BH&#39;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)&quot;,
    &quot;if(!is.null(egoLM)){
      egoLM &lt;- setReadable(egoLM, OrgDb = org.Mm.eg.db)
      dotplot(egoLM, showCategory=30)
    }&quot;,
    &quot;if(!is.null(egoLM)){
      cnetplot(egoLM, node_label=&#39;all&#39;)
    }&quot;,
    &quot;if(!is.null(egoLM)){
      emapplot(egoLM, showCategory = 15)
    }\n&quot;,
    &quot;```\n&quot;,
    &quot;\n&quot;
)
    
plots_ep &lt;- lapply(grpVec, 
  function(grp) knitr::knit_expand(text = template_ep)
)</code></pre>
<div id="bcells-2" class="section level4">
<h4>Bcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Bcells-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Bcells-1">
Past versions of “GSEA cwDE TCRM_CD45 Bcells-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Bcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Bcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Bcells-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Bcells-2">
Past versions of “GSEA cwDE TCRM_CD45 Bcells-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Bcells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Bcells-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Bcells-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Bcells-3">
Past versions of “GSEA cwDE TCRM_CD45 Bcells-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Bcells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Bcells-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dcs-2" class="section level4">
<h4>DCs</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 DCs-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-DCs-1">
Past versions of “GSEA cwDE TCRM_CD45 DCs-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-DCs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 DCs-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 DCs-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-DCs-2">
Past versions of “GSEA cwDE TCRM_CD45 DCs-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-DCs-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 DCs-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 DCs-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-DCs-3">
Past versions of “GSEA cwDE TCRM_CD45 DCs-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-DCs-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 DCs-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ilcs-1" class="section level4">
<h4>ILCs</h4>
</div>
<div id="macrophagesmonocytes-2" class="section level4">
<h4>MacrophagesMonocytes</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 MacrophagesMonocytes-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-MacrophagesMonocytes-1">
Past versions of “GSEA cwDE TCRM_CD45 MacrophagesMonocytes-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-MacrophagesMonocytes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 MacrophagesMonocytes-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 MacrophagesMonocytes-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-MacrophagesMonocytes-2">
Past versions of “GSEA cwDE TCRM_CD45 MacrophagesMonocytes-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-MacrophagesMonocytes-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 MacrophagesMonocytes-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 MacrophagesMonocytes-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-MacrophagesMonocytes-3">
Past versions of “GSEA cwDE TCRM_CD45 MacrophagesMonocytes-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-MacrophagesMonocytes-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 MacrophagesMonocytes-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="neutrophils-2" class="section level4">
<h4>Neutrophils</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Neutrophils-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Neutrophils-1">
Past versions of “GSEA cwDE TCRM_CD45 Neutrophils-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Neutrophils-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Neutrophils-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Neutrophils-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Neutrophils-2">
Past versions of “GSEA cwDE TCRM_CD45 Neutrophils-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Neutrophils-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Neutrophils-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Neutrophils-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Neutrophils-3">
Past versions of “GSEA cwDE TCRM_CD45 Neutrophils-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Neutrophils-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Neutrophils-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nkcells-2" class="section level4">
<h4>NKcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 NKcells-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-NKcells-1">
Past versions of “GSEA cwDE TCRM_CD45 NKcells-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-NKcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 NKcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 NKcells-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-NKcells-2">
Past versions of “GSEA cwDE TCRM_CD45 NKcells-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-NKcells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 NKcells-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 NKcells-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-NKcells-3">
Past versions of “GSEA cwDE TCRM_CD45 NKcells-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-NKcells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 NKcells-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="proliferating-2" class="section level4">
<h4>proliferating</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 proliferating-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-proliferating-1">
Past versions of “GSEA cwDE TCRM_CD45 proliferating-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-proliferating-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 proliferating-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 proliferating-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-proliferating-2">
Past versions of “GSEA cwDE TCRM_CD45 proliferating-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-proliferating-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 proliferating-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 proliferating-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-proliferating-3">
Past versions of “GSEA cwDE TCRM_CD45 proliferating-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-proliferating-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 proliferating-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tcells-2" class="section level4">
<h4>Tcells</h4>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Tcells-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Tcells-1">
Past versions of “GSEA cwDE TCRM_CD45 Tcells-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Tcells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Tcells-1.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Tcells-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Tcells-2">
Past versions of “GSEA cwDE TCRM_CD45 Tcells-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Tcells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Tcells-2.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Tcells-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-cwDE-TCRM_CD45-Tcells-3">
Past versions of “GSEA cwDE TCRM_CD45 Tcells-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-cwDE-TCRM_CD45-Tcells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/TCRM_Christina/blob/0435f3e9538d648a5899c21b79e3e823eb4feefb/figure/mergeCD45cells.Rmd/GSEA cwDE TCRM_CD45 Tcells-3.png" target="_blank">0435f3e</a>
</td>
<td>
mluetge
</td>
<td>
2021-06-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pheatmap_1.0.12             ggpubr_0.4.0                ggsci_2.9                  
 [4] runSeurat3_0.1.0            here_1.0.1                  magrittr_2.0.1             
 [7] SeuratObject_4.0.0          Seurat_4.0.1                forcats_0.5.1              
[10] stringr_1.4.0               dplyr_1.0.5                 purrr_0.3.4                
[13] readr_1.4.0                 tidyr_1.1.3                 tibble_3.1.1               
[16] ggplot2_3.3.3               tidyverse_1.3.1             SingleCellExperiment_1.10.1
[19] SummarizedExperiment_1.18.2 DelayedArray_0.14.1         matrixStats_0.58.0         
[22] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2         msigdbr_7.2.1              
[25] enrichplot_1.8.1            DOSE_3.14.0                 org.Mm.eg.db_3.11.4        
[28] AnnotationDbi_1.50.3        IRanges_2.22.2              S4Vectors_0.26.1           
[31] Biobase_2.48.0              BiocGenerics_0.34.0         clusterProfiler_3.16.1     

loaded via a namespace (and not attached):
  [1] scattermore_0.7        bit64_4.0.5            knitr_1.33             irlba_2.3.3           
  [5] data.table_1.14.0      rpart_4.1-15           RCurl_1.98-1.3         generics_0.1.0        
  [9] cowplot_1.1.1          RSQLite_2.2.7          RANN_2.6.1             europepmc_0.4         
 [13] future_1.21.0          bit_4.0.4              spatstat.data_2.1-0    xml2_1.3.2            
 [17] lubridate_1.7.10       httpuv_1.6.0           assertthat_0.2.1       viridis_0.6.0         
 [21] xfun_0.22              hms_1.0.0              jquerylib_0.1.4        evaluate_0.14         
 [25] promises_1.2.0.1       fansi_0.4.2            progress_1.2.2         dbplyr_2.1.1          
 [29] readxl_1.3.1           igraph_1.2.6           DBI_1.1.1              htmlwidgets_1.5.3     
 [33] spatstat.geom_2.1-0    ellipsis_0.3.1         RSpectra_0.16-0        backports_1.2.1       
 [37] deldir_0.2-10          vctrs_0.3.7            ROCR_1.0-11            abind_1.4-5           
 [41] cachem_1.0.4           withr_2.4.2            ggforce_0.3.3          triebeard_0.3.0       
 [45] sctransform_0.3.2      prettyunits_1.1.1      goftest_1.2-2          cluster_2.1.2         
 [49] lazyeval_0.2.2         crayon_1.4.1           pkgconfig_2.0.3        labeling_0.4.2        
 [53] tweenr_1.0.2           nlme_3.1-152           rlang_0.4.10           globals_0.14.0        
 [57] lifecycle_1.0.0        miniUI_0.1.1.1         downloader_0.4         modelr_0.1.8          
 [61] cellranger_1.1.0       rprojroot_2.0.2        polyclip_1.10-0        lmtest_0.9-38         
 [65] Matrix_1.3-2           urltools_1.7.3         carData_3.0-4          zoo_1.8-9             
 [69] reprex_2.0.0           whisker_0.4            ggridges_0.5.3         png_0.1-7             
 [73] viridisLite_0.4.0      bitops_1.0-7           KernSmooth_2.23-18     blob_1.2.1            
 [77] workflowr_1.6.2        qvalue_2.20.0          parallelly_1.24.0      rstatix_0.7.0         
 [81] gridGraphics_0.5-1     ggsignif_0.6.1         scales_1.1.1           memoise_2.0.0         
 [85] plyr_1.8.6             ica_1.0-2              zlibbioc_1.34.0        compiler_4.0.0        
 [89] scatterpie_0.1.6       RColorBrewer_1.1-2     fitdistrplus_1.1-3     cli_2.5.0             
 [93] XVector_0.28.0         listenv_0.8.0          patchwork_1.1.1        pbapply_1.4-3         
 [97] MASS_7.3-53.1          mgcv_1.8-35            tidyselect_1.1.0       stringi_1.5.3         
[101] highr_0.9              yaml_2.2.1             GOSemSim_2.14.2        ggrepel_0.9.1         
[105] grid_4.0.0             sass_0.3.1             fastmatch_1.1-0        tools_4.0.0           
[109] future.apply_1.7.0     rio_0.5.26             rstudioapi_0.13        foreign_0.8-81        
[113] git2r_0.28.0           gridExtra_2.3          farver_2.1.0           Rtsne_0.15            
[117] ggraph_2.0.5           digest_0.6.27          rvcheck_0.1.8          BiocManager_1.30.12   
[121] shiny_1.6.0            Rcpp_1.0.6             car_3.0-10             broom_0.7.6           
[125] later_1.2.0            RcppAnnoy_0.0.18       httr_1.4.2             colorspace_2.0-0      
[129] rvest_1.0.0            fs_1.5.0               tensor_1.5             reticulate_1.19       
[133] splines_4.0.0          uwot_0.1.10            spatstat.utils_2.1-0   graphlayouts_0.7.1    
[137] ggplotify_0.0.6        plotly_4.9.3           xtable_1.8-4           jsonlite_1.7.2        
[141] tidygraph_1.2.0        R6_2.5.0               pillar_1.6.0           htmltools_0.5.1.1     
[145] mime_0.10              glue_1.4.2             fastmap_1.1.0          BiocParallel_1.22.0   
[149] codetools_0.2-18       fgsea_1.14.0           utf8_1.2.1             lattice_0.20-41       
[153] bslib_0.2.4            spatstat.sparse_2.0-0  curl_4.3               leiden_0.3.7          
[157] zip_2.1.1              GO.db_3.11.4           openxlsx_4.2.3         limma_3.44.3          
[161] survival_3.2-11        rmarkdown_2.7          munsell_0.5.0          DO.db_2.9             
[165] GenomeInfoDbData_1.2.3 haven_2.4.1            reshape2_1.4.4         gtable_0.3.0          
[169] spatstat.core_2.1-2   </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon Jun 14 16:59:12 2021&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pheatmap_1.0.12             ggpubr_0.4.0                ggsci_2.9                  
 [4] runSeurat3_0.1.0            here_1.0.1                  magrittr_2.0.1             
 [7] SeuratObject_4.0.0          Seurat_4.0.1                forcats_0.5.1              
[10] stringr_1.4.0               dplyr_1.0.5                 purrr_0.3.4                
[13] readr_1.4.0                 tidyr_1.1.3                 tibble_3.1.1               
[16] ggplot2_3.3.3               tidyverse_1.3.1             SingleCellExperiment_1.10.1
[19] SummarizedExperiment_1.18.2 DelayedArray_0.14.1         matrixStats_0.58.0         
[22] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2         msigdbr_7.2.1              
[25] enrichplot_1.8.1            DOSE_3.14.0                 org.Mm.eg.db_3.11.4        
[28] AnnotationDbi_1.50.3        IRanges_2.22.2              S4Vectors_0.26.1           
[31] Biobase_2.48.0              BiocGenerics_0.34.0         clusterProfiler_3.16.1     

loaded via a namespace (and not attached):
  [1] scattermore_0.7        bit64_4.0.5            knitr_1.33             irlba_2.3.3           
  [5] data.table_1.14.0      rpart_4.1-15           RCurl_1.98-1.3         generics_0.1.0        
  [9] cowplot_1.1.1          RSQLite_2.2.7          RANN_2.6.1             europepmc_0.4         
 [13] future_1.21.0          bit_4.0.4              spatstat.data_2.1-0    xml2_1.3.2            
 [17] lubridate_1.7.10       httpuv_1.6.0           assertthat_0.2.1       viridis_0.6.0         
 [21] xfun_0.22              hms_1.0.0              jquerylib_0.1.4        evaluate_0.14         
 [25] promises_1.2.0.1       fansi_0.4.2            progress_1.2.2         dbplyr_2.1.1          
 [29] readxl_1.3.1           igraph_1.2.6           DBI_1.1.1              htmlwidgets_1.5.3     
 [33] spatstat.geom_2.1-0    ellipsis_0.3.1         RSpectra_0.16-0        backports_1.2.1       
 [37] deldir_0.2-10          vctrs_0.3.7            ROCR_1.0-11            abind_1.4-5           
 [41] cachem_1.0.4           withr_2.4.2            ggforce_0.3.3          triebeard_0.3.0       
 [45] sctransform_0.3.2      prettyunits_1.1.1      goftest_1.2-2          cluster_2.1.2         
 [49] lazyeval_0.2.2         crayon_1.4.1           pkgconfig_2.0.3        labeling_0.4.2        
 [53] tweenr_1.0.2           nlme_3.1-152           rlang_0.4.10           globals_0.14.0        
 [57] lifecycle_1.0.0        miniUI_0.1.1.1         downloader_0.4         modelr_0.1.8          
 [61] cellranger_1.1.0       rprojroot_2.0.2        polyclip_1.10-0        lmtest_0.9-38         
 [65] Matrix_1.3-2           urltools_1.7.3         carData_3.0-4          zoo_1.8-9             
 [69] reprex_2.0.0           whisker_0.4            ggridges_0.5.3         png_0.1-7             
 [73] viridisLite_0.4.0      bitops_1.0-7           KernSmooth_2.23-18     blob_1.2.1            
 [77] workflowr_1.6.2        qvalue_2.20.0          parallelly_1.24.0      rstatix_0.7.0         
 [81] gridGraphics_0.5-1     ggsignif_0.6.1         scales_1.1.1           memoise_2.0.0         
 [85] plyr_1.8.6             ica_1.0-2              zlibbioc_1.34.0        compiler_4.0.0        
 [89] scatterpie_0.1.6       RColorBrewer_1.1-2     fitdistrplus_1.1-3     cli_2.5.0             
 [93] XVector_0.28.0         listenv_0.8.0          patchwork_1.1.1        pbapply_1.4-3         
 [97] MASS_7.3-53.1          mgcv_1.8-35            tidyselect_1.1.0       stringi_1.5.3         
[101] highr_0.9              yaml_2.2.1             GOSemSim_2.14.2        ggrepel_0.9.1         
[105] grid_4.0.0             sass_0.3.1             fastmatch_1.1-0        tools_4.0.0           
[109] future.apply_1.7.0     rio_0.5.26             rstudioapi_0.13        foreign_0.8-81        
[113] git2r_0.28.0           gridExtra_2.3          farver_2.1.0           Rtsne_0.15            
[117] ggraph_2.0.5           digest_0.6.27          rvcheck_0.1.8          BiocManager_1.30.12   
[121] shiny_1.6.0            Rcpp_1.0.6             car_3.0-10             broom_0.7.6           
[125] later_1.2.0            RcppAnnoy_0.0.18       httr_1.4.2             colorspace_2.0-0      
[129] rvest_1.0.0            fs_1.5.0               tensor_1.5             reticulate_1.19       
[133] splines_4.0.0          uwot_0.1.10            spatstat.utils_2.1-0   graphlayouts_0.7.1    
[137] ggplotify_0.0.6        plotly_4.9.3           xtable_1.8-4           jsonlite_1.7.2        
[141] tidygraph_1.2.0        R6_2.5.0               pillar_1.6.0           htmltools_0.5.1.1     
[145] mime_0.10              glue_1.4.2             fastmap_1.1.0          BiocParallel_1.22.0   
[149] codetools_0.2-18       fgsea_1.14.0           utf8_1.2.1             lattice_0.20-41       
[153] bslib_0.2.4            spatstat.sparse_2.0-0  curl_4.3               leiden_0.3.7          
[157] zip_2.1.1              GO.db_3.11.4           openxlsx_4.2.3         limma_3.44.3          
[161] survival_3.2-11        rmarkdown_2.7          munsell_0.5.0          DO.db_2.9             
[165] GenomeInfoDbData_1.2.3 haven_2.4.1            reshape2_1.4.4         gtable_0.3.0          
[169] spatstat.core_2.1-2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
